{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f34d208f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "The following objects are masked from 'package:parallel':\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "Warning message:\n",
      "\"package 'S4Vectors' was built under R version 3.6.3\"Loading required package: stats4\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Warning message:\n",
      "\"package 'IRanges' was built under R version 3.6.2\"\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "The following object is masked from 'package:grDevices':\n",
      "\n",
      "    windows\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "Attaching package: 'Biostrings'\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    strsplit\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(Biostrings)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f71ea3fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'%in%'</li>\n",
       "\t<li>'AA_ALPHABET'</li>\n",
       "\t<li>'AA_PROTEINOGENIC'</li>\n",
       "\t<li>'AA_STANDARD'</li>\n",
       "\t<li>'AAMultipleAlignment'</li>\n",
       "\t<li>'AAString'</li>\n",
       "\t<li>'AAStringSet'</li>\n",
       "\t<li>'AAStringSetList'</li>\n",
       "\t<li>'aligned'</li>\n",
       "\t<li>'alignedPattern'</li>\n",
       "\t<li>'alignedSubject'</li>\n",
       "\t<li>'alphabet'</li>\n",
       "\t<li>'alphabetFrequency'</li>\n",
       "\t<li>'AMINO_ACID_CODE'</li>\n",
       "\t<li>'as.data.frame'</li>\n",
       "\t<li>'as.list'</li>\n",
       "\t<li>'as.matrix'</li>\n",
       "\t<li>'BString'</li>\n",
       "\t<li>'BStringSet'</li>\n",
       "\t<li>'BStringSetList'</li>\n",
       "\t<li>'chartr'</li>\n",
       "\t<li>'codons'</li>\n",
       "\t<li>'coerce'</li>\n",
       "\t<li>'collapse'</li>\n",
       "\t<li>'colmask'</li>\n",
       "\t<li>'colmask&lt;-'</li>\n",
       "\t<li>'compareStrings'</li>\n",
       "\t<li>'complement'</li>\n",
       "\t<li>'computeAllFlinks'</li>\n",
       "\t<li>'consensusMatrix'</li>\n",
       "\t<li>'consensusString'</li>\n",
       "\t<li>'consensusViews'</li>\n",
       "\t<li>'countPattern'</li>\n",
       "\t<li>'countPDict'</li>\n",
       "\t<li>'countPWM'</li>\n",
       "\t<li>'coverage'</li>\n",
       "\t<li>'deletion'</li>\n",
       "\t<li>'detail'</li>\n",
       "\t<li>'dinucleotideFrequency'</li>\n",
       "\t<li>'dinucleotideFrequencyTest'</li>\n",
       "\t<li>'DNA_ALPHABET'</li>\n",
       "\t<li>'DNA_BASES'</li>\n",
       "\t<li>'DNAMultipleAlignment'</li>\n",
       "\t<li>'DNAString'</li>\n",
       "\t<li>'DNAStringSet'</li>\n",
       "\t<li>'DNAStringSetList'</li>\n",
       "\t<li>'duplicated'</li>\n",
       "\t<li>'encoding'</li>\n",
       "\t<li>'end'</li>\n",
       "\t<li>'endIndex'</li>\n",
       "\t<li>'errorSubstitutionMatrices'</li>\n",
       "\t<li>'extract_character_from_XString_by_positions'</li>\n",
       "\t<li>'extract_character_from_XString_by_ranges'</li>\n",
       "\t<li>'extractAllMatches'</li>\n",
       "\t<li>'extractAt'</li>\n",
       "\t<li>'fasta.index'</li>\n",
       "\t<li>'fasta.seqlengths'</li>\n",
       "\t<li>'fastq.geometry'</li>\n",
       "\t<li>'fastq.seqlengths'</li>\n",
       "\t<li>'findPalindromes'</li>\n",
       "\t<li>'gaps'</li>\n",
       "\t<li>'GENETIC_CODE'</li>\n",
       "\t<li>'GENETIC_CODE_TABLE'</li>\n",
       "\t<li>'get_seqtype_conversion_lookup'</li>\n",
       "\t<li>'getGeneticCode'</li>\n",
       "\t<li>'getSeq'</li>\n",
       "\t<li>'gregexpr2'</li>\n",
       "\t<li>'hasAllFlinks'</li>\n",
       "\t<li>'hasLetterAt'</li>\n",
       "\t<li>'hasOnlyBaseLetters'</li>\n",
       "\t<li>'head'</li>\n",
       "\t<li>'IlluminaQuality'</li>\n",
       "\t<li>'indel'</li>\n",
       "\t<li>'initialize'</li>\n",
       "\t<li>'injectHardMask'</li>\n",
       "\t<li>'insertion'</li>\n",
       "\t<li>'intersect'</li>\n",
       "\t<li>'is.unsorted'</li>\n",
       "\t<li>'isMatchingAt'</li>\n",
       "\t<li>'isMatchingEndingAt'</li>\n",
       "\t<li>'isMatchingStartingAt'</li>\n",
       "\t<li>'IUPAC_CODE_MAP'</li>\n",
       "\t<li>'lcprefix'</li>\n",
       "\t<li>'lcsubstr'</li>\n",
       "\t<li>'lcsuffix'</li>\n",
       "\t<li>'letter'</li>\n",
       "\t<li>'letterFrequency'</li>\n",
       "\t<li>'letterFrequencyInSlidingView'</li>\n",
       "\t<li>'longestConsecutive'</li>\n",
       "\t<li>'make_XString_from_string'</li>\n",
       "\t<li>'make_XStringSet_from_strings'</li>\n",
       "\t<li>'mask'</li>\n",
       "\t<li>'maskeddim'</li>\n",
       "\t<li>'maskedncol'</li>\n",
       "\t<li>'maskednrow'</li>\n",
       "\t<li>'maskedratio'</li>\n",
       "\t<li>'maskedwidth'</li>\n",
       "\t<li>'maskGaps'</li>\n",
       "\t<li>'maskMotif'</li>\n",
       "\t<li>'masks'</li>\n",
       "\t<li>'masks&lt;-'</li>\n",
       "\t<li>'match'</li>\n",
       "\t<li>'matchLRPatterns'</li>\n",
       "\t<li>'matchPattern'</li>\n",
       "\t<li>'matchPDict'</li>\n",
       "\t<li>'matchProbePair'</li>\n",
       "\t<li>'matchprobes'</li>\n",
       "\t<li>'matchPWM'</li>\n",
       "\t<li>'maxScore'</li>\n",
       "\t<li>'maxWeights'</li>\n",
       "\t<li>'mergeIUPACLetters'</li>\n",
       "\t<li>'minScore'</li>\n",
       "\t<li>'minWeights'</li>\n",
       "\t<li>'mismatch'</li>\n",
       "\t<li>'mismatchSummary'</li>\n",
       "\t<li>'mismatchTable'</li>\n",
       "\t<li>'mkAllStrings'</li>\n",
       "\t<li>'N50'</li>\n",
       "\t<li>'nchar'</li>\n",
       "\t<li>'nedit'</li>\n",
       "\t<li>'neditAt'</li>\n",
       "\t<li>'neditEndingAt'</li>\n",
       "\t<li>'neditStartingAt'</li>\n",
       "\t<li>'needwunsQS'</li>\n",
       "\t<li>'nindel'</li>\n",
       "\t<li>'nmatch'</li>\n",
       "\t<li>'nmismatch'</li>\n",
       "\t<li>'nnodes'</li>\n",
       "\t<li>'nucleotideFrequencyAt'</li>\n",
       "\t<li>'nucleotideSubstitutionMatrix'</li>\n",
       "\t<li>'oligonucleotideFrequency'</li>\n",
       "\t<li>'oligonucleotideTransitions'</li>\n",
       "\t<li>'order'</li>\n",
       "\t<li>'padAndClip'</li>\n",
       "\t<li>'pairwiseAlignment'</li>\n",
       "\t<li>'PairwiseAlignments'</li>\n",
       "\t<li>'PairwiseAlignmentsSingleSubject'</li>\n",
       "\t<li>'palindromeArmLength'</li>\n",
       "\t<li>'palindromeLeftArm'</li>\n",
       "\t<li>'palindromeRightArm'</li>\n",
       "\t<li>'parallelSlotNames'</li>\n",
       "\t<li>'parallelVectorNames'</li>\n",
       "\t<li>'pattern'</li>\n",
       "\t<li>'patternFrequency'</li>\n",
       "\t<li>'pcompare'</li>\n",
       "\t<li>'PDict'</li>\n",
       "\t<li>'PhredQuality'</li>\n",
       "\t<li>'pid'</li>\n",
       "\t<li>'pmatchPattern'</li>\n",
       "\t<li>'PWM'</li>\n",
       "\t<li>'PWMscoreStartingAt'</li>\n",
       "\t<li>'quality'</li>\n",
       "\t<li>'QualityScaledAAStringSet'</li>\n",
       "\t<li>'QualityScaledBStringSet'</li>\n",
       "\t<li>'QualityScaledDNAStringSet'</li>\n",
       "\t<li>'QualityScaledRNAStringSet'</li>\n",
       "\t<li>'qualitySubstitutionMatrices'</li>\n",
       "\t<li>'rank'</li>\n",
       "\t<li>'readAAMultipleAlignment'</li>\n",
       "\t<li>'readAAStringSet'</li>\n",
       "\t<li>'readBStringSet'</li>\n",
       "\t<li>'readDNAMultipleAlignment'</li>\n",
       "\t<li>'readDNAStringSet'</li>\n",
       "\t<li>'readQualityScaledDNAStringSet'</li>\n",
       "\t<li>'readRNAMultipleAlignment'</li>\n",
       "\t<li>'readRNAStringSet'</li>\n",
       "\t<li>'relistToClass'</li>\n",
       "\t<li>'replaceAmbiguities'</li>\n",
       "\t<li>'replaceAt'</li>\n",
       "\t<li>'replaceLetterAt'</li>\n",
       "\t<li>'reverse'</li>\n",
       "\t<li>'reverseComplement'</li>\n",
       "\t<li>'RNA_ALPHABET'</li>\n",
       "\t<li>'RNA_BASES'</li>\n",
       "\t<li>'RNA_GENETIC_CODE'</li>\n",
       "\t<li>'RNAMultipleAlignment'</li>\n",
       "\t<li>'RNAString'</li>\n",
       "\t<li>'RNAStringSet'</li>\n",
       "\t<li>'RNAStringSetList'</li>\n",
       "\t<li>'rowmask'</li>\n",
       "\t<li>'rowmask&lt;-'</li>\n",
       "\t<li>'saveXStringSet'</li>\n",
       "\t<li>'score'</li>\n",
       "\t<li>'seqtype'</li>\n",
       "\t<li>'seqtype&lt;-'</li>\n",
       "\t<li>'setdiff'</li>\n",
       "\t<li>'setequal'</li>\n",
       "\t<li>'show'</li>\n",
       "\t<li>'showAsCell'</li>\n",
       "\t<li>'SolexaQuality'</li>\n",
       "\t<li>'sort'</li>\n",
       "\t<li>'stackStrings'</li>\n",
       "\t<li>'start'</li>\n",
       "\t<li>'startIndex'</li>\n",
       "\t<li>'stringDist'</li>\n",
       "\t<li>'strsplit'</li>\n",
       "\t<li>'subject'</li>\n",
       "\t<li>'subpatterns'</li>\n",
       "\t<li>'subseq'</li>\n",
       "\t<li>'subseq&lt;-'</li>\n",
       "\t<li>'substr'</li>\n",
       "\t<li>'substring'</li>\n",
       "\t<li>'summary'</li>\n",
       "\t<li>'tail'</li>\n",
       "\t<li>'tb'</li>\n",
       "\t<li>'tb.width'</li>\n",
       "\t<li>'threebands'</li>\n",
       "\t<li>'toComplex'</li>\n",
       "\t<li>'toString'</li>\n",
       "\t<li>'translate'</li>\n",
       "\t<li>'trimLRPatterns'</li>\n",
       "\t<li>'trinucleotideFrequency'</li>\n",
       "\t<li>'twoWayAlphabetFrequency'</li>\n",
       "\t<li>'type'</li>\n",
       "\t<li>'unaligned'</li>\n",
       "\t<li>'union'</li>\n",
       "\t<li>'uniqueLetters'</li>\n",
       "\t<li>'unitScale'</li>\n",
       "\t<li>'unmasked'</li>\n",
       "\t<li>'unstrsplit'</li>\n",
       "\t<li>'updateObject'</li>\n",
       "\t<li>'vcountPattern'</li>\n",
       "\t<li>'vcountPDict'</li>\n",
       "\t<li>'Views'</li>\n",
       "\t<li>'vmatchPattern'</li>\n",
       "\t<li>'vmatchPDict'</li>\n",
       "\t<li>'vwhichPDict'</li>\n",
       "\t<li>'which.isMatchingAt'</li>\n",
       "\t<li>'which.isMatchingEndingAt'</li>\n",
       "\t<li>'which.isMatchingStartingAt'</li>\n",
       "\t<li>'whichPDict'</li>\n",
       "\t<li>'width'</li>\n",
       "\t<li>'width0'</li>\n",
       "\t<li>'windows'</li>\n",
       "\t<li>'write.phylip'</li>\n",
       "\t<li>'writePairwiseAlignments'</li>\n",
       "\t<li>'writeQualityScaledXStringSet'</li>\n",
       "\t<li>'writeXStringSet'</li>\n",
       "\t<li>'xscat'</li>\n",
       "\t<li>'xscodes'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '\\%in\\%'\n",
       "\\item 'AA\\_ALPHABET'\n",
       "\\item 'AA\\_PROTEINOGENIC'\n",
       "\\item 'AA\\_STANDARD'\n",
       "\\item 'AAMultipleAlignment'\n",
       "\\item 'AAString'\n",
       "\\item 'AAStringSet'\n",
       "\\item 'AAStringSetList'\n",
       "\\item 'aligned'\n",
       "\\item 'alignedPattern'\n",
       "\\item 'alignedSubject'\n",
       "\\item 'alphabet'\n",
       "\\item 'alphabetFrequency'\n",
       "\\item 'AMINO\\_ACID\\_CODE'\n",
       "\\item 'as.data.frame'\n",
       "\\item 'as.list'\n",
       "\\item 'as.matrix'\n",
       "\\item 'BString'\n",
       "\\item 'BStringSet'\n",
       "\\item 'BStringSetList'\n",
       "\\item 'chartr'\n",
       "\\item 'codons'\n",
       "\\item 'coerce'\n",
       "\\item 'collapse'\n",
       "\\item 'colmask'\n",
       "\\item 'colmask<-'\n",
       "\\item 'compareStrings'\n",
       "\\item 'complement'\n",
       "\\item 'computeAllFlinks'\n",
       "\\item 'consensusMatrix'\n",
       "\\item 'consensusString'\n",
       "\\item 'consensusViews'\n",
       "\\item 'countPattern'\n",
       "\\item 'countPDict'\n",
       "\\item 'countPWM'\n",
       "\\item 'coverage'\n",
       "\\item 'deletion'\n",
       "\\item 'detail'\n",
       "\\item 'dinucleotideFrequency'\n",
       "\\item 'dinucleotideFrequencyTest'\n",
       "\\item 'DNA\\_ALPHABET'\n",
       "\\item 'DNA\\_BASES'\n",
       "\\item 'DNAMultipleAlignment'\n",
       "\\item 'DNAString'\n",
       "\\item 'DNAStringSet'\n",
       "\\item 'DNAStringSetList'\n",
       "\\item 'duplicated'\n",
       "\\item 'encoding'\n",
       "\\item 'end'\n",
       "\\item 'endIndex'\n",
       "\\item 'errorSubstitutionMatrices'\n",
       "\\item 'extract\\_character\\_from\\_XString\\_by\\_positions'\n",
       "\\item 'extract\\_character\\_from\\_XString\\_by\\_ranges'\n",
       "\\item 'extractAllMatches'\n",
       "\\item 'extractAt'\n",
       "\\item 'fasta.index'\n",
       "\\item 'fasta.seqlengths'\n",
       "\\item 'fastq.geometry'\n",
       "\\item 'fastq.seqlengths'\n",
       "\\item 'findPalindromes'\n",
       "\\item 'gaps'\n",
       "\\item 'GENETIC\\_CODE'\n",
       "\\item 'GENETIC\\_CODE\\_TABLE'\n",
       "\\item 'get\\_seqtype\\_conversion\\_lookup'\n",
       "\\item 'getGeneticCode'\n",
       "\\item 'getSeq'\n",
       "\\item 'gregexpr2'\n",
       "\\item 'hasAllFlinks'\n",
       "\\item 'hasLetterAt'\n",
       "\\item 'hasOnlyBaseLetters'\n",
       "\\item 'head'\n",
       "\\item 'IlluminaQuality'\n",
       "\\item 'indel'\n",
       "\\item 'initialize'\n",
       "\\item 'injectHardMask'\n",
       "\\item 'insertion'\n",
       "\\item 'intersect'\n",
       "\\item 'is.unsorted'\n",
       "\\item 'isMatchingAt'\n",
       "\\item 'isMatchingEndingAt'\n",
       "\\item 'isMatchingStartingAt'\n",
       "\\item 'IUPAC\\_CODE\\_MAP'\n",
       "\\item 'lcprefix'\n",
       "\\item 'lcsubstr'\n",
       "\\item 'lcsuffix'\n",
       "\\item 'letter'\n",
       "\\item 'letterFrequency'\n",
       "\\item 'letterFrequencyInSlidingView'\n",
       "\\item 'longestConsecutive'\n",
       "\\item 'make\\_XString\\_from\\_string'\n",
       "\\item 'make\\_XStringSet\\_from\\_strings'\n",
       "\\item 'mask'\n",
       "\\item 'maskeddim'\n",
       "\\item 'maskedncol'\n",
       "\\item 'maskednrow'\n",
       "\\item 'maskedratio'\n",
       "\\item 'maskedwidth'\n",
       "\\item 'maskGaps'\n",
       "\\item 'maskMotif'\n",
       "\\item 'masks'\n",
       "\\item 'masks<-'\n",
       "\\item 'match'\n",
       "\\item 'matchLRPatterns'\n",
       "\\item 'matchPattern'\n",
       "\\item 'matchPDict'\n",
       "\\item 'matchProbePair'\n",
       "\\item 'matchprobes'\n",
       "\\item 'matchPWM'\n",
       "\\item 'maxScore'\n",
       "\\item 'maxWeights'\n",
       "\\item 'mergeIUPACLetters'\n",
       "\\item 'minScore'\n",
       "\\item 'minWeights'\n",
       "\\item 'mismatch'\n",
       "\\item 'mismatchSummary'\n",
       "\\item 'mismatchTable'\n",
       "\\item 'mkAllStrings'\n",
       "\\item 'N50'\n",
       "\\item 'nchar'\n",
       "\\item 'nedit'\n",
       "\\item 'neditAt'\n",
       "\\item 'neditEndingAt'\n",
       "\\item 'neditStartingAt'\n",
       "\\item 'needwunsQS'\n",
       "\\item 'nindel'\n",
       "\\item 'nmatch'\n",
       "\\item 'nmismatch'\n",
       "\\item 'nnodes'\n",
       "\\item 'nucleotideFrequencyAt'\n",
       "\\item 'nucleotideSubstitutionMatrix'\n",
       "\\item 'oligonucleotideFrequency'\n",
       "\\item 'oligonucleotideTransitions'\n",
       "\\item 'order'\n",
       "\\item 'padAndClip'\n",
       "\\item 'pairwiseAlignment'\n",
       "\\item 'PairwiseAlignments'\n",
       "\\item 'PairwiseAlignmentsSingleSubject'\n",
       "\\item 'palindromeArmLength'\n",
       "\\item 'palindromeLeftArm'\n",
       "\\item 'palindromeRightArm'\n",
       "\\item 'parallelSlotNames'\n",
       "\\item 'parallelVectorNames'\n",
       "\\item 'pattern'\n",
       "\\item 'patternFrequency'\n",
       "\\item 'pcompare'\n",
       "\\item 'PDict'\n",
       "\\item 'PhredQuality'\n",
       "\\item 'pid'\n",
       "\\item 'pmatchPattern'\n",
       "\\item 'PWM'\n",
       "\\item 'PWMscoreStartingAt'\n",
       "\\item 'quality'\n",
       "\\item 'QualityScaledAAStringSet'\n",
       "\\item 'QualityScaledBStringSet'\n",
       "\\item 'QualityScaledDNAStringSet'\n",
       "\\item 'QualityScaledRNAStringSet'\n",
       "\\item 'qualitySubstitutionMatrices'\n",
       "\\item 'rank'\n",
       "\\item 'readAAMultipleAlignment'\n",
       "\\item 'readAAStringSet'\n",
       "\\item 'readBStringSet'\n",
       "\\item 'readDNAMultipleAlignment'\n",
       "\\item 'readDNAStringSet'\n",
       "\\item 'readQualityScaledDNAStringSet'\n",
       "\\item 'readRNAMultipleAlignment'\n",
       "\\item 'readRNAStringSet'\n",
       "\\item 'relistToClass'\n",
       "\\item 'replaceAmbiguities'\n",
       "\\item 'replaceAt'\n",
       "\\item 'replaceLetterAt'\n",
       "\\item 'reverse'\n",
       "\\item 'reverseComplement'\n",
       "\\item 'RNA\\_ALPHABET'\n",
       "\\item 'RNA\\_BASES'\n",
       "\\item 'RNA\\_GENETIC\\_CODE'\n",
       "\\item 'RNAMultipleAlignment'\n",
       "\\item 'RNAString'\n",
       "\\item 'RNAStringSet'\n",
       "\\item 'RNAStringSetList'\n",
       "\\item 'rowmask'\n",
       "\\item 'rowmask<-'\n",
       "\\item 'saveXStringSet'\n",
       "\\item 'score'\n",
       "\\item 'seqtype'\n",
       "\\item 'seqtype<-'\n",
       "\\item 'setdiff'\n",
       "\\item 'setequal'\n",
       "\\item 'show'\n",
       "\\item 'showAsCell'\n",
       "\\item 'SolexaQuality'\n",
       "\\item 'sort'\n",
       "\\item 'stackStrings'\n",
       "\\item 'start'\n",
       "\\item 'startIndex'\n",
       "\\item 'stringDist'\n",
       "\\item 'strsplit'\n",
       "\\item 'subject'\n",
       "\\item 'subpatterns'\n",
       "\\item 'subseq'\n",
       "\\item 'subseq<-'\n",
       "\\item 'substr'\n",
       "\\item 'substring'\n",
       "\\item 'summary'\n",
       "\\item 'tail'\n",
       "\\item 'tb'\n",
       "\\item 'tb.width'\n",
       "\\item 'threebands'\n",
       "\\item 'toComplex'\n",
       "\\item 'toString'\n",
       "\\item 'translate'\n",
       "\\item 'trimLRPatterns'\n",
       "\\item 'trinucleotideFrequency'\n",
       "\\item 'twoWayAlphabetFrequency'\n",
       "\\item 'type'\n",
       "\\item 'unaligned'\n",
       "\\item 'union'\n",
       "\\item 'uniqueLetters'\n",
       "\\item 'unitScale'\n",
       "\\item 'unmasked'\n",
       "\\item 'unstrsplit'\n",
       "\\item 'updateObject'\n",
       "\\item 'vcountPattern'\n",
       "\\item 'vcountPDict'\n",
       "\\item 'Views'\n",
       "\\item 'vmatchPattern'\n",
       "\\item 'vmatchPDict'\n",
       "\\item 'vwhichPDict'\n",
       "\\item 'which.isMatchingAt'\n",
       "\\item 'which.isMatchingEndingAt'\n",
       "\\item 'which.isMatchingStartingAt'\n",
       "\\item 'whichPDict'\n",
       "\\item 'width'\n",
       "\\item 'width0'\n",
       "\\item 'windows'\n",
       "\\item 'write.phylip'\n",
       "\\item 'writePairwiseAlignments'\n",
       "\\item 'writeQualityScaledXStringSet'\n",
       "\\item 'writeXStringSet'\n",
       "\\item 'xscat'\n",
       "\\item 'xscodes'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '%in%'\n",
       "2. 'AA_ALPHABET'\n",
       "3. 'AA_PROTEINOGENIC'\n",
       "4. 'AA_STANDARD'\n",
       "5. 'AAMultipleAlignment'\n",
       "6. 'AAString'\n",
       "7. 'AAStringSet'\n",
       "8. 'AAStringSetList'\n",
       "9. 'aligned'\n",
       "10. 'alignedPattern'\n",
       "11. 'alignedSubject'\n",
       "12. 'alphabet'\n",
       "13. 'alphabetFrequency'\n",
       "14. 'AMINO_ACID_CODE'\n",
       "15. 'as.data.frame'\n",
       "16. 'as.list'\n",
       "17. 'as.matrix'\n",
       "18. 'BString'\n",
       "19. 'BStringSet'\n",
       "20. 'BStringSetList'\n",
       "21. 'chartr'\n",
       "22. 'codons'\n",
       "23. 'coerce'\n",
       "24. 'collapse'\n",
       "25. 'colmask'\n",
       "26. 'colmask&lt;-'\n",
       "27. 'compareStrings'\n",
       "28. 'complement'\n",
       "29. 'computeAllFlinks'\n",
       "30. 'consensusMatrix'\n",
       "31. 'consensusString'\n",
       "32. 'consensusViews'\n",
       "33. 'countPattern'\n",
       "34. 'countPDict'\n",
       "35. 'countPWM'\n",
       "36. 'coverage'\n",
       "37. 'deletion'\n",
       "38. 'detail'\n",
       "39. 'dinucleotideFrequency'\n",
       "40. 'dinucleotideFrequencyTest'\n",
       "41. 'DNA_ALPHABET'\n",
       "42. 'DNA_BASES'\n",
       "43. 'DNAMultipleAlignment'\n",
       "44. 'DNAString'\n",
       "45. 'DNAStringSet'\n",
       "46. 'DNAStringSetList'\n",
       "47. 'duplicated'\n",
       "48. 'encoding'\n",
       "49. 'end'\n",
       "50. 'endIndex'\n",
       "51. 'errorSubstitutionMatrices'\n",
       "52. 'extract_character_from_XString_by_positions'\n",
       "53. 'extract_character_from_XString_by_ranges'\n",
       "54. 'extractAllMatches'\n",
       "55. 'extractAt'\n",
       "56. 'fasta.index'\n",
       "57. 'fasta.seqlengths'\n",
       "58. 'fastq.geometry'\n",
       "59. 'fastq.seqlengths'\n",
       "60. 'findPalindromes'\n",
       "61. 'gaps'\n",
       "62. 'GENETIC_CODE'\n",
       "63. 'GENETIC_CODE_TABLE'\n",
       "64. 'get_seqtype_conversion_lookup'\n",
       "65. 'getGeneticCode'\n",
       "66. 'getSeq'\n",
       "67. 'gregexpr2'\n",
       "68. 'hasAllFlinks'\n",
       "69. 'hasLetterAt'\n",
       "70. 'hasOnlyBaseLetters'\n",
       "71. 'head'\n",
       "72. 'IlluminaQuality'\n",
       "73. 'indel'\n",
       "74. 'initialize'\n",
       "75. 'injectHardMask'\n",
       "76. 'insertion'\n",
       "77. 'intersect'\n",
       "78. 'is.unsorted'\n",
       "79. 'isMatchingAt'\n",
       "80. 'isMatchingEndingAt'\n",
       "81. 'isMatchingStartingAt'\n",
       "82. 'IUPAC_CODE_MAP'\n",
       "83. 'lcprefix'\n",
       "84. 'lcsubstr'\n",
       "85. 'lcsuffix'\n",
       "86. 'letter'\n",
       "87. 'letterFrequency'\n",
       "88. 'letterFrequencyInSlidingView'\n",
       "89. 'longestConsecutive'\n",
       "90. 'make_XString_from_string'\n",
       "91. 'make_XStringSet_from_strings'\n",
       "92. 'mask'\n",
       "93. 'maskeddim'\n",
       "94. 'maskedncol'\n",
       "95. 'maskednrow'\n",
       "96. 'maskedratio'\n",
       "97. 'maskedwidth'\n",
       "98. 'maskGaps'\n",
       "99. 'maskMotif'\n",
       "100. 'masks'\n",
       "101. 'masks&lt;-'\n",
       "102. 'match'\n",
       "103. 'matchLRPatterns'\n",
       "104. 'matchPattern'\n",
       "105. 'matchPDict'\n",
       "106. 'matchProbePair'\n",
       "107. 'matchprobes'\n",
       "108. 'matchPWM'\n",
       "109. 'maxScore'\n",
       "110. 'maxWeights'\n",
       "111. 'mergeIUPACLetters'\n",
       "112. 'minScore'\n",
       "113. 'minWeights'\n",
       "114. 'mismatch'\n",
       "115. 'mismatchSummary'\n",
       "116. 'mismatchTable'\n",
       "117. 'mkAllStrings'\n",
       "118. 'N50'\n",
       "119. 'nchar'\n",
       "120. 'nedit'\n",
       "121. 'neditAt'\n",
       "122. 'neditEndingAt'\n",
       "123. 'neditStartingAt'\n",
       "124. 'needwunsQS'\n",
       "125. 'nindel'\n",
       "126. 'nmatch'\n",
       "127. 'nmismatch'\n",
       "128. 'nnodes'\n",
       "129. 'nucleotideFrequencyAt'\n",
       "130. 'nucleotideSubstitutionMatrix'\n",
       "131. 'oligonucleotideFrequency'\n",
       "132. 'oligonucleotideTransitions'\n",
       "133. 'order'\n",
       "134. 'padAndClip'\n",
       "135. 'pairwiseAlignment'\n",
       "136. 'PairwiseAlignments'\n",
       "137. 'PairwiseAlignmentsSingleSubject'\n",
       "138. 'palindromeArmLength'\n",
       "139. 'palindromeLeftArm'\n",
       "140. 'palindromeRightArm'\n",
       "141. 'parallelSlotNames'\n",
       "142. 'parallelVectorNames'\n",
       "143. 'pattern'\n",
       "144. 'patternFrequency'\n",
       "145. 'pcompare'\n",
       "146. 'PDict'\n",
       "147. 'PhredQuality'\n",
       "148. 'pid'\n",
       "149. 'pmatchPattern'\n",
       "150. 'PWM'\n",
       "151. 'PWMscoreStartingAt'\n",
       "152. 'quality'\n",
       "153. 'QualityScaledAAStringSet'\n",
       "154. 'QualityScaledBStringSet'\n",
       "155. 'QualityScaledDNAStringSet'\n",
       "156. 'QualityScaledRNAStringSet'\n",
       "157. 'qualitySubstitutionMatrices'\n",
       "158. 'rank'\n",
       "159. 'readAAMultipleAlignment'\n",
       "160. 'readAAStringSet'\n",
       "161. 'readBStringSet'\n",
       "162. 'readDNAMultipleAlignment'\n",
       "163. 'readDNAStringSet'\n",
       "164. 'readQualityScaledDNAStringSet'\n",
       "165. 'readRNAMultipleAlignment'\n",
       "166. 'readRNAStringSet'\n",
       "167. 'relistToClass'\n",
       "168. 'replaceAmbiguities'\n",
       "169. 'replaceAt'\n",
       "170. 'replaceLetterAt'\n",
       "171. 'reverse'\n",
       "172. 'reverseComplement'\n",
       "173. 'RNA_ALPHABET'\n",
       "174. 'RNA_BASES'\n",
       "175. 'RNA_GENETIC_CODE'\n",
       "176. 'RNAMultipleAlignment'\n",
       "177. 'RNAString'\n",
       "178. 'RNAStringSet'\n",
       "179. 'RNAStringSetList'\n",
       "180. 'rowmask'\n",
       "181. 'rowmask&lt;-'\n",
       "182. 'saveXStringSet'\n",
       "183. 'score'\n",
       "184. 'seqtype'\n",
       "185. 'seqtype&lt;-'\n",
       "186. 'setdiff'\n",
       "187. 'setequal'\n",
       "188. 'show'\n",
       "189. 'showAsCell'\n",
       "190. 'SolexaQuality'\n",
       "191. 'sort'\n",
       "192. 'stackStrings'\n",
       "193. 'start'\n",
       "194. 'startIndex'\n",
       "195. 'stringDist'\n",
       "196. 'strsplit'\n",
       "197. 'subject'\n",
       "198. 'subpatterns'\n",
       "199. 'subseq'\n",
       "200. 'subseq&lt;-'\n",
       "201. 'substr'\n",
       "202. 'substring'\n",
       "203. 'summary'\n",
       "204. 'tail'\n",
       "205. 'tb'\n",
       "206. 'tb.width'\n",
       "207. 'threebands'\n",
       "208. 'toComplex'\n",
       "209. 'toString'\n",
       "210. 'translate'\n",
       "211. 'trimLRPatterns'\n",
       "212. 'trinucleotideFrequency'\n",
       "213. 'twoWayAlphabetFrequency'\n",
       "214. 'type'\n",
       "215. 'unaligned'\n",
       "216. 'union'\n",
       "217. 'uniqueLetters'\n",
       "218. 'unitScale'\n",
       "219. 'unmasked'\n",
       "220. 'unstrsplit'\n",
       "221. 'updateObject'\n",
       "222. 'vcountPattern'\n",
       "223. 'vcountPDict'\n",
       "224. 'Views'\n",
       "225. 'vmatchPattern'\n",
       "226. 'vmatchPDict'\n",
       "227. 'vwhichPDict'\n",
       "228. 'which.isMatchingAt'\n",
       "229. 'which.isMatchingEndingAt'\n",
       "230. 'which.isMatchingStartingAt'\n",
       "231. 'whichPDict'\n",
       "232. 'width'\n",
       "233. 'width0'\n",
       "234. 'windows'\n",
       "235. 'write.phylip'\n",
       "236. 'writePairwiseAlignments'\n",
       "237. 'writeQualityScaledXStringSet'\n",
       "238. 'writeXStringSet'\n",
       "239. 'xscat'\n",
       "240. 'xscodes'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"%in%\"                                       \n",
       "  [2] \"AA_ALPHABET\"                                \n",
       "  [3] \"AA_PROTEINOGENIC\"                           \n",
       "  [4] \"AA_STANDARD\"                                \n",
       "  [5] \"AAMultipleAlignment\"                        \n",
       "  [6] \"AAString\"                                   \n",
       "  [7] \"AAStringSet\"                                \n",
       "  [8] \"AAStringSetList\"                            \n",
       "  [9] \"aligned\"                                    \n",
       " [10] \"alignedPattern\"                             \n",
       " [11] \"alignedSubject\"                             \n",
       " [12] \"alphabet\"                                   \n",
       " [13] \"alphabetFrequency\"                          \n",
       " [14] \"AMINO_ACID_CODE\"                            \n",
       " [15] \"as.data.frame\"                              \n",
       " [16] \"as.list\"                                    \n",
       " [17] \"as.matrix\"                                  \n",
       " [18] \"BString\"                                    \n",
       " [19] \"BStringSet\"                                 \n",
       " [20] \"BStringSetList\"                             \n",
       " [21] \"chartr\"                                     \n",
       " [22] \"codons\"                                     \n",
       " [23] \"coerce\"                                     \n",
       " [24] \"collapse\"                                   \n",
       " [25] \"colmask\"                                    \n",
       " [26] \"colmask<-\"                                  \n",
       " [27] \"compareStrings\"                             \n",
       " [28] \"complement\"                                 \n",
       " [29] \"computeAllFlinks\"                           \n",
       " [30] \"consensusMatrix\"                            \n",
       " [31] \"consensusString\"                            \n",
       " [32] \"consensusViews\"                             \n",
       " [33] \"countPattern\"                               \n",
       " [34] \"countPDict\"                                 \n",
       " [35] \"countPWM\"                                   \n",
       " [36] \"coverage\"                                   \n",
       " [37] \"deletion\"                                   \n",
       " [38] \"detail\"                                     \n",
       " [39] \"dinucleotideFrequency\"                      \n",
       " [40] \"dinucleotideFrequencyTest\"                  \n",
       " [41] \"DNA_ALPHABET\"                               \n",
       " [42] \"DNA_BASES\"                                  \n",
       " [43] \"DNAMultipleAlignment\"                       \n",
       " [44] \"DNAString\"                                  \n",
       " [45] \"DNAStringSet\"                               \n",
       " [46] \"DNAStringSetList\"                           \n",
       " [47] \"duplicated\"                                 \n",
       " [48] \"encoding\"                                   \n",
       " [49] \"end\"                                        \n",
       " [50] \"endIndex\"                                   \n",
       " [51] \"errorSubstitutionMatrices\"                  \n",
       " [52] \"extract_character_from_XString_by_positions\"\n",
       " [53] \"extract_character_from_XString_by_ranges\"   \n",
       " [54] \"extractAllMatches\"                          \n",
       " [55] \"extractAt\"                                  \n",
       " [56] \"fasta.index\"                                \n",
       " [57] \"fasta.seqlengths\"                           \n",
       " [58] \"fastq.geometry\"                             \n",
       " [59] \"fastq.seqlengths\"                           \n",
       " [60] \"findPalindromes\"                            \n",
       " [61] \"gaps\"                                       \n",
       " [62] \"GENETIC_CODE\"                               \n",
       " [63] \"GENETIC_CODE_TABLE\"                         \n",
       " [64] \"get_seqtype_conversion_lookup\"              \n",
       " [65] \"getGeneticCode\"                             \n",
       " [66] \"getSeq\"                                     \n",
       " [67] \"gregexpr2\"                                  \n",
       " [68] \"hasAllFlinks\"                               \n",
       " [69] \"hasLetterAt\"                                \n",
       " [70] \"hasOnlyBaseLetters\"                         \n",
       " [71] \"head\"                                       \n",
       " [72] \"IlluminaQuality\"                            \n",
       " [73] \"indel\"                                      \n",
       " [74] \"initialize\"                                 \n",
       " [75] \"injectHardMask\"                             \n",
       " [76] \"insertion\"                                  \n",
       " [77] \"intersect\"                                  \n",
       " [78] \"is.unsorted\"                                \n",
       " [79] \"isMatchingAt\"                               \n",
       " [80] \"isMatchingEndingAt\"                         \n",
       " [81] \"isMatchingStartingAt\"                       \n",
       " [82] \"IUPAC_CODE_MAP\"                             \n",
       " [83] \"lcprefix\"                                   \n",
       " [84] \"lcsubstr\"                                   \n",
       " [85] \"lcsuffix\"                                   \n",
       " [86] \"letter\"                                     \n",
       " [87] \"letterFrequency\"                            \n",
       " [88] \"letterFrequencyInSlidingView\"               \n",
       " [89] \"longestConsecutive\"                         \n",
       " [90] \"make_XString_from_string\"                   \n",
       " [91] \"make_XStringSet_from_strings\"               \n",
       " [92] \"mask\"                                       \n",
       " [93] \"maskeddim\"                                  \n",
       " [94] \"maskedncol\"                                 \n",
       " [95] \"maskednrow\"                                 \n",
       " [96] \"maskedratio\"                                \n",
       " [97] \"maskedwidth\"                                \n",
       " [98] \"maskGaps\"                                   \n",
       " [99] \"maskMotif\"                                  \n",
       "[100] \"masks\"                                      \n",
       "[101] \"masks<-\"                                    \n",
       "[102] \"match\"                                      \n",
       "[103] \"matchLRPatterns\"                            \n",
       "[104] \"matchPattern\"                               \n",
       "[105] \"matchPDict\"                                 \n",
       "[106] \"matchProbePair\"                             \n",
       "[107] \"matchprobes\"                                \n",
       "[108] \"matchPWM\"                                   \n",
       "[109] \"maxScore\"                                   \n",
       "[110] \"maxWeights\"                                 \n",
       "[111] \"mergeIUPACLetters\"                          \n",
       "[112] \"minScore\"                                   \n",
       "[113] \"minWeights\"                                 \n",
       "[114] \"mismatch\"                                   \n",
       "[115] \"mismatchSummary\"                            \n",
       "[116] \"mismatchTable\"                              \n",
       "[117] \"mkAllStrings\"                               \n",
       "[118] \"N50\"                                        \n",
       "[119] \"nchar\"                                      \n",
       "[120] \"nedit\"                                      \n",
       "[121] \"neditAt\"                                    \n",
       "[122] \"neditEndingAt\"                              \n",
       "[123] \"neditStartingAt\"                            \n",
       "[124] \"needwunsQS\"                                 \n",
       "[125] \"nindel\"                                     \n",
       "[126] \"nmatch\"                                     \n",
       "[127] \"nmismatch\"                                  \n",
       "[128] \"nnodes\"                                     \n",
       "[129] \"nucleotideFrequencyAt\"                      \n",
       "[130] \"nucleotideSubstitutionMatrix\"               \n",
       "[131] \"oligonucleotideFrequency\"                   \n",
       "[132] \"oligonucleotideTransitions\"                 \n",
       "[133] \"order\"                                      \n",
       "[134] \"padAndClip\"                                 \n",
       "[135] \"pairwiseAlignment\"                          \n",
       "[136] \"PairwiseAlignments\"                         \n",
       "[137] \"PairwiseAlignmentsSingleSubject\"            \n",
       "[138] \"palindromeArmLength\"                        \n",
       "[139] \"palindromeLeftArm\"                          \n",
       "[140] \"palindromeRightArm\"                         \n",
       "[141] \"parallelSlotNames\"                          \n",
       "[142] \"parallelVectorNames\"                        \n",
       "[143] \"pattern\"                                    \n",
       "[144] \"patternFrequency\"                           \n",
       "[145] \"pcompare\"                                   \n",
       "[146] \"PDict\"                                      \n",
       "[147] \"PhredQuality\"                               \n",
       "[148] \"pid\"                                        \n",
       "[149] \"pmatchPattern\"                              \n",
       "[150] \"PWM\"                                        \n",
       "[151] \"PWMscoreStartingAt\"                         \n",
       "[152] \"quality\"                                    \n",
       "[153] \"QualityScaledAAStringSet\"                   \n",
       "[154] \"QualityScaledBStringSet\"                    \n",
       "[155] \"QualityScaledDNAStringSet\"                  \n",
       "[156] \"QualityScaledRNAStringSet\"                  \n",
       "[157] \"qualitySubstitutionMatrices\"                \n",
       "[158] \"rank\"                                       \n",
       "[159] \"readAAMultipleAlignment\"                    \n",
       "[160] \"readAAStringSet\"                            \n",
       "[161] \"readBStringSet\"                             \n",
       "[162] \"readDNAMultipleAlignment\"                   \n",
       "[163] \"readDNAStringSet\"                           \n",
       "[164] \"readQualityScaledDNAStringSet\"              \n",
       "[165] \"readRNAMultipleAlignment\"                   \n",
       "[166] \"readRNAStringSet\"                           \n",
       "[167] \"relistToClass\"                              \n",
       "[168] \"replaceAmbiguities\"                         \n",
       "[169] \"replaceAt\"                                  \n",
       "[170] \"replaceLetterAt\"                            \n",
       "[171] \"reverse\"                                    \n",
       "[172] \"reverseComplement\"                          \n",
       "[173] \"RNA_ALPHABET\"                               \n",
       "[174] \"RNA_BASES\"                                  \n",
       "[175] \"RNA_GENETIC_CODE\"                           \n",
       "[176] \"RNAMultipleAlignment\"                       \n",
       "[177] \"RNAString\"                                  \n",
       "[178] \"RNAStringSet\"                               \n",
       "[179] \"RNAStringSetList\"                           \n",
       "[180] \"rowmask\"                                    \n",
       "[181] \"rowmask<-\"                                  \n",
       "[182] \"saveXStringSet\"                             \n",
       "[183] \"score\"                                      \n",
       "[184] \"seqtype\"                                    \n",
       "[185] \"seqtype<-\"                                  \n",
       "[186] \"setdiff\"                                    \n",
       "[187] \"setequal\"                                   \n",
       "[188] \"show\"                                       \n",
       "[189] \"showAsCell\"                                 \n",
       "[190] \"SolexaQuality\"                              \n",
       "[191] \"sort\"                                       \n",
       "[192] \"stackStrings\"                               \n",
       "[193] \"start\"                                      \n",
       "[194] \"startIndex\"                                 \n",
       "[195] \"stringDist\"                                 \n",
       "[196] \"strsplit\"                                   \n",
       "[197] \"subject\"                                    \n",
       "[198] \"subpatterns\"                                \n",
       "[199] \"subseq\"                                     \n",
       "[200] \"subseq<-\"                                   \n",
       "[201] \"substr\"                                     \n",
       "[202] \"substring\"                                  \n",
       "[203] \"summary\"                                    \n",
       "[204] \"tail\"                                       \n",
       "[205] \"tb\"                                         \n",
       "[206] \"tb.width\"                                   \n",
       "[207] \"threebands\"                                 \n",
       "[208] \"toComplex\"                                  \n",
       "[209] \"toString\"                                   \n",
       "[210] \"translate\"                                  \n",
       "[211] \"trimLRPatterns\"                             \n",
       "[212] \"trinucleotideFrequency\"                     \n",
       "[213] \"twoWayAlphabetFrequency\"                    \n",
       "[214] \"type\"                                       \n",
       "[215] \"unaligned\"                                  \n",
       "[216] \"union\"                                      \n",
       "[217] \"uniqueLetters\"                              \n",
       "[218] \"unitScale\"                                  \n",
       "[219] \"unmasked\"                                   \n",
       "[220] \"unstrsplit\"                                 \n",
       "[221] \"updateObject\"                               \n",
       "[222] \"vcountPattern\"                              \n",
       "[223] \"vcountPDict\"                                \n",
       "[224] \"Views\"                                      \n",
       "[225] \"vmatchPattern\"                              \n",
       "[226] \"vmatchPDict\"                                \n",
       "[227] \"vwhichPDict\"                                \n",
       "[228] \"which.isMatchingAt\"                         \n",
       "[229] \"which.isMatchingEndingAt\"                   \n",
       "[230] \"which.isMatchingStartingAt\"                 \n",
       "[231] \"whichPDict\"                                 \n",
       "[232] \"width\"                                      \n",
       "[233] \"width0\"                                     \n",
       "[234] \"windows\"                                    \n",
       "[235] \"write.phylip\"                               \n",
       "[236] \"writePairwiseAlignments\"                    \n",
       "[237] \"writeQualityScaledXStringSet\"               \n",
       "[238] \"writeXStringSet\"                            \n",
       "[239] \"xscat\"                                      \n",
       "[240] \"xscodes\"                                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls(\"package:Biostrings\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "34c0eb2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'A'</li>\n",
       "\t<li>'R'</li>\n",
       "\t<li>'N'</li>\n",
       "\t<li>'D'</li>\n",
       "\t<li>'C'</li>\n",
       "\t<li>'Q'</li>\n",
       "\t<li>'E'</li>\n",
       "\t<li>'G'</li>\n",
       "\t<li>'H'</li>\n",
       "\t<li>'I'</li>\n",
       "\t<li>'L'</li>\n",
       "\t<li>'K'</li>\n",
       "\t<li>'M'</li>\n",
       "\t<li>'F'</li>\n",
       "\t<li>'P'</li>\n",
       "\t<li>'S'</li>\n",
       "\t<li>'T'</li>\n",
       "\t<li>'W'</li>\n",
       "\t<li>'Y'</li>\n",
       "\t<li>'V'</li>\n",
       "\t<li>'U'</li>\n",
       "\t<li>'O'</li>\n",
       "\t<li>'B'</li>\n",
       "\t<li>'J'</li>\n",
       "\t<li>'Z'</li>\n",
       "\t<li>'X'</li>\n",
       "\t<li>'*'</li>\n",
       "\t<li>'-'</li>\n",
       "\t<li>'+'</li>\n",
       "\t<li>'.'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A'\n",
       "\\item 'R'\n",
       "\\item 'N'\n",
       "\\item 'D'\n",
       "\\item 'C'\n",
       "\\item 'Q'\n",
       "\\item 'E'\n",
       "\\item 'G'\n",
       "\\item 'H'\n",
       "\\item 'I'\n",
       "\\item 'L'\n",
       "\\item 'K'\n",
       "\\item 'M'\n",
       "\\item 'F'\n",
       "\\item 'P'\n",
       "\\item 'S'\n",
       "\\item 'T'\n",
       "\\item 'W'\n",
       "\\item 'Y'\n",
       "\\item 'V'\n",
       "\\item 'U'\n",
       "\\item 'O'\n",
       "\\item 'B'\n",
       "\\item 'J'\n",
       "\\item 'Z'\n",
       "\\item 'X'\n",
       "\\item '*'\n",
       "\\item '-'\n",
       "\\item '+'\n",
       "\\item '.'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A'\n",
       "2. 'R'\n",
       "3. 'N'\n",
       "4. 'D'\n",
       "5. 'C'\n",
       "6. 'Q'\n",
       "7. 'E'\n",
       "8. 'G'\n",
       "9. 'H'\n",
       "10. 'I'\n",
       "11. 'L'\n",
       "12. 'K'\n",
       "13. 'M'\n",
       "14. 'F'\n",
       "15. 'P'\n",
       "16. 'S'\n",
       "17. 'T'\n",
       "18. 'W'\n",
       "19. 'Y'\n",
       "20. 'V'\n",
       "21. 'U'\n",
       "22. 'O'\n",
       "23. 'B'\n",
       "24. 'J'\n",
       "25. 'Z'\n",
       "26. 'X'\n",
       "27. '*'\n",
       "28. '-'\n",
       "29. '+'\n",
       "30. '.'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"A\" \"R\" \"N\" \"D\" \"C\" \"Q\" \"E\" \"G\" \"H\" \"I\" \"L\" \"K\" \"M\" \"F\" \"P\" \"S\" \"T\" \"W\" \"Y\"\n",
       "[20] \"V\" \"U\" \"O\" \"B\" \"J\" \"Z\" \"X\" \"*\" \"-\" \"+\" \".\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "AA_ALPHABET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d7b85ef3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'A'</li>\n",
       "\t<li>'R'</li>\n",
       "\t<li>'N'</li>\n",
       "\t<li>'D'</li>\n",
       "\t<li>'C'</li>\n",
       "\t<li>'Q'</li>\n",
       "\t<li>'E'</li>\n",
       "\t<li>'G'</li>\n",
       "\t<li>'H'</li>\n",
       "\t<li>'I'</li>\n",
       "\t<li>'L'</li>\n",
       "\t<li>'K'</li>\n",
       "\t<li>'M'</li>\n",
       "\t<li>'F'</li>\n",
       "\t<li>'P'</li>\n",
       "\t<li>'S'</li>\n",
       "\t<li>'T'</li>\n",
       "\t<li>'W'</li>\n",
       "\t<li>'Y'</li>\n",
       "\t<li>'V'</li>\n",
       "\t<li>'U'</li>\n",
       "\t<li>'O'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A'\n",
       "\\item 'R'\n",
       "\\item 'N'\n",
       "\\item 'D'\n",
       "\\item 'C'\n",
       "\\item 'Q'\n",
       "\\item 'E'\n",
       "\\item 'G'\n",
       "\\item 'H'\n",
       "\\item 'I'\n",
       "\\item 'L'\n",
       "\\item 'K'\n",
       "\\item 'M'\n",
       "\\item 'F'\n",
       "\\item 'P'\n",
       "\\item 'S'\n",
       "\\item 'T'\n",
       "\\item 'W'\n",
       "\\item 'Y'\n",
       "\\item 'V'\n",
       "\\item 'U'\n",
       "\\item 'O'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A'\n",
       "2. 'R'\n",
       "3. 'N'\n",
       "4. 'D'\n",
       "5. 'C'\n",
       "6. 'Q'\n",
       "7. 'E'\n",
       "8. 'G'\n",
       "9. 'H'\n",
       "10. 'I'\n",
       "11. 'L'\n",
       "12. 'K'\n",
       "13. 'M'\n",
       "14. 'F'\n",
       "15. 'P'\n",
       "16. 'S'\n",
       "17. 'T'\n",
       "18. 'W'\n",
       "19. 'Y'\n",
       "20. 'V'\n",
       "21. 'U'\n",
       "22. 'O'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"A\" \"R\" \"N\" \"D\" \"C\" \"Q\" \"E\" \"G\" \"H\" \"I\" \"L\" \"K\" \"M\" \"F\" \"P\" \"S\" \"T\" \"W\" \"Y\"\n",
       "[20] \"V\" \"U\" \"O\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "AA_PROTEINOGENIC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1dd13951",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'A'</li>\n",
       "\t<li>'R'</li>\n",
       "\t<li>'N'</li>\n",
       "\t<li>'D'</li>\n",
       "\t<li>'C'</li>\n",
       "\t<li>'Q'</li>\n",
       "\t<li>'E'</li>\n",
       "\t<li>'G'</li>\n",
       "\t<li>'H'</li>\n",
       "\t<li>'I'</li>\n",
       "\t<li>'L'</li>\n",
       "\t<li>'K'</li>\n",
       "\t<li>'M'</li>\n",
       "\t<li>'F'</li>\n",
       "\t<li>'P'</li>\n",
       "\t<li>'S'</li>\n",
       "\t<li>'T'</li>\n",
       "\t<li>'W'</li>\n",
       "\t<li>'Y'</li>\n",
       "\t<li>'V'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A'\n",
       "\\item 'R'\n",
       "\\item 'N'\n",
       "\\item 'D'\n",
       "\\item 'C'\n",
       "\\item 'Q'\n",
       "\\item 'E'\n",
       "\\item 'G'\n",
       "\\item 'H'\n",
       "\\item 'I'\n",
       "\\item 'L'\n",
       "\\item 'K'\n",
       "\\item 'M'\n",
       "\\item 'F'\n",
       "\\item 'P'\n",
       "\\item 'S'\n",
       "\\item 'T'\n",
       "\\item 'W'\n",
       "\\item 'Y'\n",
       "\\item 'V'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A'\n",
       "2. 'R'\n",
       "3. 'N'\n",
       "4. 'D'\n",
       "5. 'C'\n",
       "6. 'Q'\n",
       "7. 'E'\n",
       "8. 'G'\n",
       "9. 'H'\n",
       "10. 'I'\n",
       "11. 'L'\n",
       "12. 'K'\n",
       "13. 'M'\n",
       "14. 'F'\n",
       "15. 'P'\n",
       "16. 'S'\n",
       "17. 'T'\n",
       "18. 'W'\n",
       "19. 'Y'\n",
       "20. 'V'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"A\" \"R\" \"N\" \"D\" \"C\" \"Q\" \"E\" \"G\" \"H\" \"I\" \"L\" \"K\" \"M\" \"F\" \"P\" \"S\" \"T\" \"W\" \"Y\"\n",
       "[20] \"V\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "AA_STANDARD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "58b9c9b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>A</dt>\n",
       "\t\t<dd>'Ala'</dd>\n",
       "\t<dt>R</dt>\n",
       "\t\t<dd>'Arg'</dd>\n",
       "\t<dt>N</dt>\n",
       "\t\t<dd>'Asn'</dd>\n",
       "\t<dt>D</dt>\n",
       "\t\t<dd>'Asp'</dd>\n",
       "\t<dt>C</dt>\n",
       "\t\t<dd>'Cys'</dd>\n",
       "\t<dt>Q</dt>\n",
       "\t\t<dd>'Gln'</dd>\n",
       "\t<dt>E</dt>\n",
       "\t\t<dd>'Glu'</dd>\n",
       "\t<dt>G</dt>\n",
       "\t\t<dd>'Gly'</dd>\n",
       "\t<dt>H</dt>\n",
       "\t\t<dd>'His'</dd>\n",
       "\t<dt>I</dt>\n",
       "\t\t<dd>'Ile'</dd>\n",
       "\t<dt>L</dt>\n",
       "\t\t<dd>'Leu'</dd>\n",
       "\t<dt>K</dt>\n",
       "\t\t<dd>'Lys'</dd>\n",
       "\t<dt>M</dt>\n",
       "\t\t<dd>'Met'</dd>\n",
       "\t<dt>F</dt>\n",
       "\t\t<dd>'Phe'</dd>\n",
       "\t<dt>P</dt>\n",
       "\t\t<dd>'Pro'</dd>\n",
       "\t<dt>S</dt>\n",
       "\t\t<dd>'Ser'</dd>\n",
       "\t<dt>T</dt>\n",
       "\t\t<dd>'Thr'</dd>\n",
       "\t<dt>W</dt>\n",
       "\t\t<dd>'Trp'</dd>\n",
       "\t<dt>Y</dt>\n",
       "\t\t<dd>'Tyr'</dd>\n",
       "\t<dt>V</dt>\n",
       "\t\t<dd>'Val'</dd>\n",
       "\t<dt>U</dt>\n",
       "\t\t<dd>'Sec'</dd>\n",
       "\t<dt>O</dt>\n",
       "\t\t<dd>'Pyl'</dd>\n",
       "\t<dt>B</dt>\n",
       "\t\t<dd>'Asx'</dd>\n",
       "\t<dt>J</dt>\n",
       "\t\t<dd>'Xle'</dd>\n",
       "\t<dt>Z</dt>\n",
       "\t\t<dd>'Glx'</dd>\n",
       "\t<dt>X</dt>\n",
       "\t\t<dd>'Xaa'</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[A] 'Ala'\n",
       "\\item[R] 'Arg'\n",
       "\\item[N] 'Asn'\n",
       "\\item[D] 'Asp'\n",
       "\\item[C] 'Cys'\n",
       "\\item[Q] 'Gln'\n",
       "\\item[E] 'Glu'\n",
       "\\item[G] 'Gly'\n",
       "\\item[H] 'His'\n",
       "\\item[I] 'Ile'\n",
       "\\item[L] 'Leu'\n",
       "\\item[K] 'Lys'\n",
       "\\item[M] 'Met'\n",
       "\\item[F] 'Phe'\n",
       "\\item[P] 'Pro'\n",
       "\\item[S] 'Ser'\n",
       "\\item[T] 'Thr'\n",
       "\\item[W] 'Trp'\n",
       "\\item[Y] 'Tyr'\n",
       "\\item[V] 'Val'\n",
       "\\item[U] 'Sec'\n",
       "\\item[O] 'Pyl'\n",
       "\\item[B] 'Asx'\n",
       "\\item[J] 'Xle'\n",
       "\\item[Z] 'Glx'\n",
       "\\item[X] 'Xaa'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "A\n",
       ":   'Ala'R\n",
       ":   'Arg'N\n",
       ":   'Asn'D\n",
       ":   'Asp'C\n",
       ":   'Cys'Q\n",
       ":   'Gln'E\n",
       ":   'Glu'G\n",
       ":   'Gly'H\n",
       ":   'His'I\n",
       ":   'Ile'L\n",
       ":   'Leu'K\n",
       ":   'Lys'M\n",
       ":   'Met'F\n",
       ":   'Phe'P\n",
       ":   'Pro'S\n",
       ":   'Ser'T\n",
       ":   'Thr'W\n",
       ":   'Trp'Y\n",
       ":   'Tyr'V\n",
       ":   'Val'U\n",
       ":   'Sec'O\n",
       ":   'Pyl'B\n",
       ":   'Asx'J\n",
       ":   'Xle'Z\n",
       ":   'Glx'X\n",
       ":   'Xaa'\n",
       "\n"
      ],
      "text/plain": [
       "    A     R     N     D     C     Q     E     G     H     I     L     K     M \n",
       "\"Ala\" \"Arg\" \"Asn\" \"Asp\" \"Cys\" \"Gln\" \"Glu\" \"Gly\" \"His\" \"Ile\" \"Leu\" \"Lys\" \"Met\" \n",
       "    F     P     S     T     W     Y     V     U     O     B     J     Z     X \n",
       "\"Phe\" \"Pro\" \"Ser\" \"Thr\" \"Trp\" \"Tyr\" \"Val\" \"Sec\" \"Pyl\" \"Asx\" \"Xle\" \"Glx\" \"Xaa\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "AMINO_ACID_CODE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "06ff86c6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'A'</li>\n",
       "\t<li>'C'</li>\n",
       "\t<li>'G'</li>\n",
       "\t<li>'T'</li>\n",
       "\t<li>'M'</li>\n",
       "\t<li>'R'</li>\n",
       "\t<li>'W'</li>\n",
       "\t<li>'S'</li>\n",
       "\t<li>'Y'</li>\n",
       "\t<li>'K'</li>\n",
       "\t<li>'V'</li>\n",
       "\t<li>'H'</li>\n",
       "\t<li>'D'</li>\n",
       "\t<li>'B'</li>\n",
       "\t<li>'N'</li>\n",
       "\t<li>'-'</li>\n",
       "\t<li>'+'</li>\n",
       "\t<li>'.'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A'\n",
       "\\item 'C'\n",
       "\\item 'G'\n",
       "\\item 'T'\n",
       "\\item 'M'\n",
       "\\item 'R'\n",
       "\\item 'W'\n",
       "\\item 'S'\n",
       "\\item 'Y'\n",
       "\\item 'K'\n",
       "\\item 'V'\n",
       "\\item 'H'\n",
       "\\item 'D'\n",
       "\\item 'B'\n",
       "\\item 'N'\n",
       "\\item '-'\n",
       "\\item '+'\n",
       "\\item '.'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A'\n",
       "2. 'C'\n",
       "3. 'G'\n",
       "4. 'T'\n",
       "5. 'M'\n",
       "6. 'R'\n",
       "7. 'W'\n",
       "8. 'S'\n",
       "9. 'Y'\n",
       "10. 'K'\n",
       "11. 'V'\n",
       "12. 'H'\n",
       "13. 'D'\n",
       "14. 'B'\n",
       "15. 'N'\n",
       "16. '-'\n",
       "17. '+'\n",
       "18. '.'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"A\" \"C\" \"G\" \"T\" \"M\" \"R\" \"W\" \"S\" \"Y\" \"K\" \"V\" \"H\" \"D\" \"B\" \"N\" \"-\" \"+\" \".\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "DNA_ALPHABET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d624450a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'A'</li>\n",
       "\t<li>'C'</li>\n",
       "\t<li>'G'</li>\n",
       "\t<li>'T'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A'\n",
       "\\item 'C'\n",
       "\\item 'G'\n",
       "\\item 'T'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A'\n",
       "2. 'C'\n",
       "3. 'G'\n",
       "4. 'T'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"A\" \"C\" \"G\" \"T\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "DNA_BASES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "b4f88f82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>TTT</dt>\n",
       "\t\t<dd>'F'</dd>\n",
       "\t<dt>TTC</dt>\n",
       "\t\t<dd>'F'</dd>\n",
       "\t<dt>TTA</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>TTG</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>TCT</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>TCC</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>TCA</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>TCG</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>TAT</dt>\n",
       "\t\t<dd>'Y'</dd>\n",
       "\t<dt>TAC</dt>\n",
       "\t\t<dd>'Y'</dd>\n",
       "\t<dt>TAA</dt>\n",
       "\t\t<dd>'*'</dd>\n",
       "\t<dt>TAG</dt>\n",
       "\t\t<dd>'*'</dd>\n",
       "\t<dt>TGT</dt>\n",
       "\t\t<dd>'C'</dd>\n",
       "\t<dt>TGC</dt>\n",
       "\t\t<dd>'C'</dd>\n",
       "\t<dt>TGA</dt>\n",
       "\t\t<dd>'*'</dd>\n",
       "\t<dt>TGG</dt>\n",
       "\t\t<dd>'W'</dd>\n",
       "\t<dt>CTT</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>CTC</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>CTA</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>CTG</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>CCT</dt>\n",
       "\t\t<dd>'P'</dd>\n",
       "\t<dt>CCC</dt>\n",
       "\t\t<dd>'P'</dd>\n",
       "\t<dt>CCA</dt>\n",
       "\t\t<dd>'P'</dd>\n",
       "\t<dt>CCG</dt>\n",
       "\t\t<dd>'P'</dd>\n",
       "\t<dt>CAT</dt>\n",
       "\t\t<dd>'H'</dd>\n",
       "\t<dt>CAC</dt>\n",
       "\t\t<dd>'H'</dd>\n",
       "\t<dt>CAA</dt>\n",
       "\t\t<dd>'Q'</dd>\n",
       "\t<dt>CAG</dt>\n",
       "\t\t<dd>'Q'</dd>\n",
       "\t<dt>CGT</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>CGC</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>CGA</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>CGG</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>ATT</dt>\n",
       "\t\t<dd>'I'</dd>\n",
       "\t<dt>ATC</dt>\n",
       "\t\t<dd>'I'</dd>\n",
       "\t<dt>ATA</dt>\n",
       "\t\t<dd>'I'</dd>\n",
       "\t<dt>ATG</dt>\n",
       "\t\t<dd>'M'</dd>\n",
       "\t<dt>ACT</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>ACC</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>ACA</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>ACG</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>AAT</dt>\n",
       "\t\t<dd>'N'</dd>\n",
       "\t<dt>AAC</dt>\n",
       "\t\t<dd>'N'</dd>\n",
       "\t<dt>AAA</dt>\n",
       "\t\t<dd>'K'</dd>\n",
       "\t<dt>AAG</dt>\n",
       "\t\t<dd>'K'</dd>\n",
       "\t<dt>AGT</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>AGC</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>AGA</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>AGG</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>GTT</dt>\n",
       "\t\t<dd>'V'</dd>\n",
       "\t<dt>GTC</dt>\n",
       "\t\t<dd>'V'</dd>\n",
       "\t<dt>GTA</dt>\n",
       "\t\t<dd>'V'</dd>\n",
       "\t<dt>GTG</dt>\n",
       "\t\t<dd>'V'</dd>\n",
       "\t<dt>GCT</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>GCC</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>GCA</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>GCG</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>GAT</dt>\n",
       "\t\t<dd>'D'</dd>\n",
       "\t<dt>GAC</dt>\n",
       "\t\t<dd>'D'</dd>\n",
       "\t<dt>GAA</dt>\n",
       "\t\t<dd>'E'</dd>\n",
       "\t<dt>GAG</dt>\n",
       "\t\t<dd>'E'</dd>\n",
       "\t<dt>GGT</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "\t<dt>GGC</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "\t<dt>GGA</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "\t<dt>GGG</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[TTT] 'F'\n",
       "\\item[TTC] 'F'\n",
       "\\item[TTA] 'L'\n",
       "\\item[TTG] 'L'\n",
       "\\item[TCT] 'S'\n",
       "\\item[TCC] 'S'\n",
       "\\item[TCA] 'S'\n",
       "\\item[TCG] 'S'\n",
       "\\item[TAT] 'Y'\n",
       "\\item[TAC] 'Y'\n",
       "\\item[TAA] '*'\n",
       "\\item[TAG] '*'\n",
       "\\item[TGT] 'C'\n",
       "\\item[TGC] 'C'\n",
       "\\item[TGA] '*'\n",
       "\\item[TGG] 'W'\n",
       "\\item[CTT] 'L'\n",
       "\\item[CTC] 'L'\n",
       "\\item[CTA] 'L'\n",
       "\\item[CTG] 'L'\n",
       "\\item[CCT] 'P'\n",
       "\\item[CCC] 'P'\n",
       "\\item[CCA] 'P'\n",
       "\\item[CCG] 'P'\n",
       "\\item[CAT] 'H'\n",
       "\\item[CAC] 'H'\n",
       "\\item[CAA] 'Q'\n",
       "\\item[CAG] 'Q'\n",
       "\\item[CGT] 'R'\n",
       "\\item[CGC] 'R'\n",
       "\\item[CGA] 'R'\n",
       "\\item[CGG] 'R'\n",
       "\\item[ATT] 'I'\n",
       "\\item[ATC] 'I'\n",
       "\\item[ATA] 'I'\n",
       "\\item[ATG] 'M'\n",
       "\\item[ACT] 'T'\n",
       "\\item[ACC] 'T'\n",
       "\\item[ACA] 'T'\n",
       "\\item[ACG] 'T'\n",
       "\\item[AAT] 'N'\n",
       "\\item[AAC] 'N'\n",
       "\\item[AAA] 'K'\n",
       "\\item[AAG] 'K'\n",
       "\\item[AGT] 'S'\n",
       "\\item[AGC] 'S'\n",
       "\\item[AGA] 'R'\n",
       "\\item[AGG] 'R'\n",
       "\\item[GTT] 'V'\n",
       "\\item[GTC] 'V'\n",
       "\\item[GTA] 'V'\n",
       "\\item[GTG] 'V'\n",
       "\\item[GCT] 'A'\n",
       "\\item[GCC] 'A'\n",
       "\\item[GCA] 'A'\n",
       "\\item[GCG] 'A'\n",
       "\\item[GAT] 'D'\n",
       "\\item[GAC] 'D'\n",
       "\\item[GAA] 'E'\n",
       "\\item[GAG] 'E'\n",
       "\\item[GGT] 'G'\n",
       "\\item[GGC] 'G'\n",
       "\\item[GGA] 'G'\n",
       "\\item[GGG] 'G'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "TTT\n",
       ":   'F'TTC\n",
       ":   'F'TTA\n",
       ":   'L'TTG\n",
       ":   'L'TCT\n",
       ":   'S'TCC\n",
       ":   'S'TCA\n",
       ":   'S'TCG\n",
       ":   'S'TAT\n",
       ":   'Y'TAC\n",
       ":   'Y'TAA\n",
       ":   '*'TAG\n",
       ":   '*'TGT\n",
       ":   'C'TGC\n",
       ":   'C'TGA\n",
       ":   '*'TGG\n",
       ":   'W'CTT\n",
       ":   'L'CTC\n",
       ":   'L'CTA\n",
       ":   'L'CTG\n",
       ":   'L'CCT\n",
       ":   'P'CCC\n",
       ":   'P'CCA\n",
       ":   'P'CCG\n",
       ":   'P'CAT\n",
       ":   'H'CAC\n",
       ":   'H'CAA\n",
       ":   'Q'CAG\n",
       ":   'Q'CGT\n",
       ":   'R'CGC\n",
       ":   'R'CGA\n",
       ":   'R'CGG\n",
       ":   'R'ATT\n",
       ":   'I'ATC\n",
       ":   'I'ATA\n",
       ":   'I'ATG\n",
       ":   'M'ACT\n",
       ":   'T'ACC\n",
       ":   'T'ACA\n",
       ":   'T'ACG\n",
       ":   'T'AAT\n",
       ":   'N'AAC\n",
       ":   'N'AAA\n",
       ":   'K'AAG\n",
       ":   'K'AGT\n",
       ":   'S'AGC\n",
       ":   'S'AGA\n",
       ":   'R'AGG\n",
       ":   'R'GTT\n",
       ":   'V'GTC\n",
       ":   'V'GTA\n",
       ":   'V'GTG\n",
       ":   'V'GCT\n",
       ":   'A'GCC\n",
       ":   'A'GCA\n",
       ":   'A'GCG\n",
       ":   'A'GAT\n",
       ":   'D'GAC\n",
       ":   'D'GAA\n",
       ":   'E'GAG\n",
       ":   'E'GGT\n",
       ":   'G'GGC\n",
       ":   'G'GGA\n",
       ":   'G'GGG\n",
       ":   'G'\n",
       "\n"
      ],
      "text/plain": [
       "TTT TTC TTA TTG TCT TCC TCA TCG TAT TAC TAA TAG TGT TGC TGA TGG CTT CTC CTA CTG \n",
       "\"F\" \"F\" \"L\" \"L\" \"S\" \"S\" \"S\" \"S\" \"Y\" \"Y\" \"*\" \"*\" \"C\" \"C\" \"*\" \"W\" \"L\" \"L\" \"L\" \"L\" \n",
       "CCT CCC CCA CCG CAT CAC CAA CAG CGT CGC CGA CGG ATT ATC ATA ATG ACT ACC ACA ACG \n",
       "\"P\" \"P\" \"P\" \"P\" \"H\" \"H\" \"Q\" \"Q\" \"R\" \"R\" \"R\" \"R\" \"I\" \"I\" \"I\" \"M\" \"T\" \"T\" \"T\" \"T\" \n",
       "AAT AAC AAA AAG AGT AGC AGA AGG GTT GTC GTA GTG GCT GCC GCA GCG GAT GAC GAA GAG \n",
       "\"N\" \"N\" \"K\" \"K\" \"S\" \"S\" \"R\" \"R\" \"V\" \"V\" \"V\" \"V\" \"A\" \"A\" \"A\" \"A\" \"D\" \"D\" \"E\" \"E\" \n",
       "GGT GGC GGA GGG \n",
       "\"G\" \"G\" \"G\" \"G\" \n",
       "attr(,\"alt_init_codons\")\n",
       "[1] \"TTG\" \"CTG\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "GENETIC_CODE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "59053e6f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>name</th><th scope=col>name2</th><th scope=col>id</th><th scope=col>AAs</th><th scope=col>Starts</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>Standard                                                                                        </td><td>SGC0                                                                                            </td><td>1                                                                                               </td><td>FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>---M---------------M---------------M----------------------------                                </td></tr>\n",
       "\t<tr><td>Vertebrate Mitochondrial                                                                        </td><td>SGC1                                                                                            </td><td>2                                                                                               </td><td>FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSS**VVVVAAAADDEEGGGG                                </td><td>--------------------------------MMMM---------------M------------                                </td></tr>\n",
       "\t<tr><td>Yeast Mitochondrial                                                                             </td><td>SGC2                                                                                            </td><td>3                                                                                               </td><td>FFLLSSSSYY**CCWWTTTTPPPPHHQQRRRRIIMMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>----------------------------------MM----------------------------                                </td></tr>\n",
       "\t<tr><td>Mold Mitochondrial; Protozoan Mitochondrial; Coelenterate Mitochondrial; Mycoplasma; Spiroplasma</td><td>SGC3                                                                                            </td><td>4                                                                                               </td><td>FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>--MM---------------M------------MMMM---------------M------------                                </td></tr>\n",
       "\t<tr><td>Invertebrate Mitochondrial                                                                      </td><td>SGC4                                                                                            </td><td>5                                                                                               </td><td>FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSSSVVVVAAAADDEEGGGG                                </td><td>---M----------------------------MMMM---------------M------------                                </td></tr>\n",
       "\t<tr><td>Ciliate Nuclear; Dasycladacean Nuclear; Hexamita Nuclear                                        </td><td>SGC5                                                                                            </td><td>6                                                                                               </td><td>FFLLSSSSYYQQCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>-----------------------------------M----------------------------                                </td></tr>\n",
       "\t<tr><td>Echinoderm Mitochondrial; Flatworm Mitochondrial                                                </td><td>SGC8                                                                                            </td><td>9                                                                                               </td><td>FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                </td><td>-----------------------------------M---------------M------------                                </td></tr>\n",
       "\t<tr><td>Euplotid Nuclear                                                                                </td><td>SGC9                                                                                            </td><td>10                                                                                              </td><td>FFLLSSSSYY**CCCWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>-----------------------------------M----------------------------                                </td></tr>\n",
       "\t<tr><td>Bacterial, Archaeal and Plant Plastid                                                           </td><td>NA                                                                                              </td><td>11                                                                                              </td><td>FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>---M---------------M------------MMMM---------------M------------                                </td></tr>\n",
       "\t<tr><td>Alternative Yeast Nuclear                                                                       </td><td>NA                                                                                              </td><td>12                                                                                              </td><td>FFLLSSSSYY**CC*WLLLSPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>-------------------M---------------M----------------------------                                </td></tr>\n",
       "\t<tr><td>Ascidian Mitochondrial                                                                          </td><td>NA                                                                                              </td><td>13                                                                                              </td><td>FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSGGVVVVAAAADDEEGGGG                                </td><td>---M------------------------------MM---------------M------------                                </td></tr>\n",
       "\t<tr><td>Alternative Flatworm Mitochondrial                                                              </td><td>NA                                                                                              </td><td>14                                                                                              </td><td>FFLLSSSSYYY*CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                </td><td>-----------------------------------M----------------------------                                </td></tr>\n",
       "\t<tr><td>Blepharisma Macronuclear                                                                        </td><td>NA                                                                                              </td><td>15                                                                                              </td><td>FFLLSSSSYY*QCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>-----------------------------------M----------------------------                                </td></tr>\n",
       "\t<tr><td>Chlorophycean Mitochondrial                                                                     </td><td>NA                                                                                              </td><td>16                                                                                              </td><td>FFLLSSSSYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>-----------------------------------M----------------------------                                </td></tr>\n",
       "\t<tr><td>Trematode Mitochondrial                                                                         </td><td>NA                                                                                              </td><td>21                                                                                              </td><td>FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                </td><td>-----------------------------------M---------------M------------                                </td></tr>\n",
       "\t<tr><td>Scenedesmus obliquus Mitochondrial                                                              </td><td>NA                                                                                              </td><td>22                                                                                              </td><td>FFLLSS*SYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>-----------------------------------M----------------------------                                </td></tr>\n",
       "\t<tr><td>Thraustochytrium Mitochondrial                                                                  </td><td>NA                                                                                              </td><td>23                                                                                              </td><td>FF*LSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>--------------------------------M--M---------------M------------                                </td></tr>\n",
       "\t<tr><td>Pterobranchia Mitochondrial                                                                     </td><td>NA                                                                                              </td><td>24                                                                                              </td><td>FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSSKVVVVAAAADDEEGGGG                                </td><td>---M---------------M---------------M---------------M------------                                </td></tr>\n",
       "\t<tr><td>Candidate Division SR1 and Gracilibacteria                                                      </td><td>NA                                                                                              </td><td>25                                                                                              </td><td>FFLLSSSSYY**CCGWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>---M-------------------------------M---------------M------------                                </td></tr>\n",
       "\t<tr><td>Pachysolen tannophilus Nuclear                                                                  </td><td>NA                                                                                              </td><td>26                                                                                              </td><td>FFLLSSSSYY**CC*WLLLAPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                </td><td>-------------------M---------------M----------------------------                                </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllll}\n",
       " name & name2 & id & AAs & Starts\\\\\n",
       "\\hline\n",
       "\t Standard                                                                                         & SGC0                                                                                             & 1                                                                                                & FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & ---M---------------M---------------M----------------------------                                \\\\\n",
       "\t Vertebrate Mitochondrial                                                                         & SGC1                                                                                             & 2                                                                                                & FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSS**VVVVAAAADDEEGGGG                                 & --------------------------------MMMM---------------M------------                                \\\\\n",
       "\t Yeast Mitochondrial                                                                              & SGC2                                                                                             & 3                                                                                                & FFLLSSSSYY**CCWWTTTTPPPPHHQQRRRRIIMMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & ----------------------------------MM----------------------------                                \\\\\n",
       "\t Mold Mitochondrial; Protozoan Mitochondrial; Coelenterate Mitochondrial; Mycoplasma; Spiroplasma & SGC3                                                                                             & 4                                                                                                & FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & --MM---------------M------------MMMM---------------M------------                                \\\\\n",
       "\t Invertebrate Mitochondrial                                                                       & SGC4                                                                                             & 5                                                                                                & FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSSSVVVVAAAADDEEGGGG                                 & ---M----------------------------MMMM---------------M------------                                \\\\\n",
       "\t Ciliate Nuclear; Dasycladacean Nuclear; Hexamita Nuclear                                         & SGC5                                                                                             & 6                                                                                                & FFLLSSSSYYQQCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & -----------------------------------M----------------------------                                \\\\\n",
       "\t Echinoderm Mitochondrial; Flatworm Mitochondrial                                                 & SGC8                                                                                             & 9                                                                                                & FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                 & -----------------------------------M---------------M------------                                \\\\\n",
       "\t Euplotid Nuclear                                                                                 & SGC9                                                                                             & 10                                                                                               & FFLLSSSSYY**CCCWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & -----------------------------------M----------------------------                                \\\\\n",
       "\t Bacterial, Archaeal and Plant Plastid                                                            & NA                                                                                               & 11                                                                                               & FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & ---M---------------M------------MMMM---------------M------------                                \\\\\n",
       "\t Alternative Yeast Nuclear                                                                        & NA                                                                                               & 12                                                                                               & FFLLSSSSYY**CC*WLLLSPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & -------------------M---------------M----------------------------                                \\\\\n",
       "\t Ascidian Mitochondrial                                                                           & NA                                                                                               & 13                                                                                               & FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSGGVVVVAAAADDEEGGGG                                 & ---M------------------------------MM---------------M------------                                \\\\\n",
       "\t Alternative Flatworm Mitochondrial                                                               & NA                                                                                               & 14                                                                                               & FFLLSSSSYYY*CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                 & -----------------------------------M----------------------------                                \\\\\n",
       "\t Blepharisma Macronuclear                                                                         & NA                                                                                               & 15                                                                                               & FFLLSSSSYY*QCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & -----------------------------------M----------------------------                                \\\\\n",
       "\t Chlorophycean Mitochondrial                                                                      & NA                                                                                               & 16                                                                                               & FFLLSSSSYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & -----------------------------------M----------------------------                                \\\\\n",
       "\t Trematode Mitochondrial                                                                          & NA                                                                                               & 21                                                                                               & FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                 & -----------------------------------M---------------M------------                                \\\\\n",
       "\t Scenedesmus obliquus Mitochondrial                                                               & NA                                                                                               & 22                                                                                               & FFLLSS*SYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & -----------------------------------M----------------------------                                \\\\\n",
       "\t Thraustochytrium Mitochondrial                                                                   & NA                                                                                               & 23                                                                                               & FF*LSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & --------------------------------M--M---------------M------------                                \\\\\n",
       "\t Pterobranchia Mitochondrial                                                                      & NA                                                                                               & 24                                                                                               & FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSSKVVVVAAAADDEEGGGG                                 & ---M---------------M---------------M---------------M------------                                \\\\\n",
       "\t Candidate Division SR1 and Gracilibacteria                                                       & NA                                                                                               & 25                                                                                               & FFLLSSSSYY**CCGWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & ---M-------------------------------M---------------M------------                                \\\\\n",
       "\t Pachysolen tannophilus Nuclear                                                                   & NA                                                                                               & 26                                                                                               & FFLLSSSSYY**CC*WLLLAPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 & -------------------M---------------M----------------------------                                \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| name | name2 | id | AAs | Starts |\n",
       "|---|---|---|---|---|\n",
       "| Standard                                                                                         | SGC0                                                                                             | 1                                                                                                | FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | ---M---------------M---------------M----------------------------                                 |\n",
       "| Vertebrate Mitochondrial                                                                         | SGC1                                                                                             | 2                                                                                                | FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSS**VVVVAAAADDEEGGGG                                 | --------------------------------MMMM---------------M------------                                 |\n",
       "| Yeast Mitochondrial                                                                              | SGC2                                                                                             | 3                                                                                                | FFLLSSSSYY**CCWWTTTTPPPPHHQQRRRRIIMMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | ----------------------------------MM----------------------------                                 |\n",
       "| Mold Mitochondrial; Protozoan Mitochondrial; Coelenterate Mitochondrial; Mycoplasma; Spiroplasma | SGC3                                                                                             | 4                                                                                                | FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | --MM---------------M------------MMMM---------------M------------                                 |\n",
       "| Invertebrate Mitochondrial                                                                       | SGC4                                                                                             | 5                                                                                                | FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSSSVVVVAAAADDEEGGGG                                 | ---M----------------------------MMMM---------------M------------                                 |\n",
       "| Ciliate Nuclear; Dasycladacean Nuclear; Hexamita Nuclear                                         | SGC5                                                                                             | 6                                                                                                | FFLLSSSSYYQQCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | -----------------------------------M----------------------------                                 |\n",
       "| Echinoderm Mitochondrial; Flatworm Mitochondrial                                                 | SGC8                                                                                             | 9                                                                                                | FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                 | -----------------------------------M---------------M------------                                 |\n",
       "| Euplotid Nuclear                                                                                 | SGC9                                                                                             | 10                                                                                               | FFLLSSSSYY**CCCWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | -----------------------------------M----------------------------                                 |\n",
       "| Bacterial, Archaeal and Plant Plastid                                                            | NA                                                                                               | 11                                                                                               | FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | ---M---------------M------------MMMM---------------M------------                                 |\n",
       "| Alternative Yeast Nuclear                                                                        | NA                                                                                               | 12                                                                                               | FFLLSSSSYY**CC*WLLLSPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | -------------------M---------------M----------------------------                                 |\n",
       "| Ascidian Mitochondrial                                                                           | NA                                                                                               | 13                                                                                               | FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSGGVVVVAAAADDEEGGGG                                 | ---M------------------------------MM---------------M------------                                 |\n",
       "| Alternative Flatworm Mitochondrial                                                               | NA                                                                                               | 14                                                                                               | FFLLSSSSYYY*CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                 | -----------------------------------M----------------------------                                 |\n",
       "| Blepharisma Macronuclear                                                                         | NA                                                                                               | 15                                                                                               | FFLLSSSSYY*QCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | -----------------------------------M----------------------------                                 |\n",
       "| Chlorophycean Mitochondrial                                                                      | NA                                                                                               | 16                                                                                               | FFLLSSSSYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | -----------------------------------M----------------------------                                 |\n",
       "| Trematode Mitochondrial                                                                          | NA                                                                                               | 21                                                                                               | FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNNKSSSSVVVVAAAADDEEGGGG                                 | -----------------------------------M---------------M------------                                 |\n",
       "| Scenedesmus obliquus Mitochondrial                                                               | NA                                                                                               | 22                                                                                               | FFLLSS*SYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | -----------------------------------M----------------------------                                 |\n",
       "| Thraustochytrium Mitochondrial                                                                   | NA                                                                                               | 23                                                                                               | FF*LSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | --------------------------------M--M---------------M------------                                 |\n",
       "| Pterobranchia Mitochondrial                                                                      | NA                                                                                               | 24                                                                                               | FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSSKVVVVAAAADDEEGGGG                                 | ---M---------------M---------------M---------------M------------                                 |\n",
       "| Candidate Division SR1 and Gracilibacteria                                                       | NA                                                                                               | 25                                                                                               | FFLLSSSSYY**CCGWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | ---M-------------------------------M---------------M------------                                 |\n",
       "| Pachysolen tannophilus Nuclear                                                                   | NA                                                                                               | 26                                                                                               | FFLLSSSSYY**CC*WLLLAPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG                                 | -------------------M---------------M----------------------------                                 |\n",
       "\n"
      ],
      "text/plain": [
       "   name                                                                                            \n",
       "1  Standard                                                                                        \n",
       "2  Vertebrate Mitochondrial                                                                        \n",
       "3  Yeast Mitochondrial                                                                             \n",
       "4  Mold Mitochondrial; Protozoan Mitochondrial; Coelenterate Mitochondrial; Mycoplasma; Spiroplasma\n",
       "5  Invertebrate Mitochondrial                                                                      \n",
       "6  Ciliate Nuclear; Dasycladacean Nuclear; Hexamita Nuclear                                        \n",
       "7  Echinoderm Mitochondrial; Flatworm Mitochondrial                                                \n",
       "8  Euplotid Nuclear                                                                                \n",
       "9  Bacterial, Archaeal and Plant Plastid                                                           \n",
       "10 Alternative Yeast Nuclear                                                                       \n",
       "11 Ascidian Mitochondrial                                                                          \n",
       "12 Alternative Flatworm Mitochondrial                                                              \n",
       "13 Blepharisma Macronuclear                                                                        \n",
       "14 Chlorophycean Mitochondrial                                                                     \n",
       "15 Trematode Mitochondrial                                                                         \n",
       "16 Scenedesmus obliquus Mitochondrial                                                              \n",
       "17 Thraustochytrium Mitochondrial                                                                  \n",
       "18 Pterobranchia Mitochondrial                                                                     \n",
       "19 Candidate Division SR1 and Gracilibacteria                                                      \n",
       "20 Pachysolen tannophilus Nuclear                                                                  \n",
       "   name2 id AAs                                                             \n",
       "1  SGC0  1  FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "2  SGC1  2  FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSS**VVVVAAAADDEEGGGG\n",
       "3  SGC2  3  FFLLSSSSYY**CCWWTTTTPPPPHHQQRRRRIIMMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "4  SGC3  4  FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "5  SGC4  5  FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSSSVVVVAAAADDEEGGGG\n",
       "6  SGC5  6  FFLLSSSSYYQQCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "7  SGC8  9  FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG\n",
       "8  SGC9  10 FFLLSSSSYY**CCCWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "9  NA    11 FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "10 NA    12 FFLLSSSSYY**CC*WLLLSPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "11 NA    13 FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNKKSSGGVVVVAAAADDEEGGGG\n",
       "12 NA    14 FFLLSSSSYYY*CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNNKSSSSVVVVAAAADDEEGGGG\n",
       "13 NA    15 FFLLSSSSYY*QCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "14 NA    16 FFLLSSSSYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "15 NA    21 FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIMMTTTTNNNKSSSSVVVVAAAADDEEGGGG\n",
       "16 NA    22 FFLLSS*SYY*LCC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "17 NA    23 FF*LSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "18 NA    24 FFLLSSSSYY**CCWWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSSKVVVVAAAADDEEGGGG\n",
       "19 NA    25 FFLLSSSSYY**CCGWLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "20 NA    26 FFLLSSSSYY**CC*WLLLAPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG\n",
       "   Starts                                                          \n",
       "1  ---M---------------M---------------M----------------------------\n",
       "2  --------------------------------MMMM---------------M------------\n",
       "3  ----------------------------------MM----------------------------\n",
       "4  --MM---------------M------------MMMM---------------M------------\n",
       "5  ---M----------------------------MMMM---------------M------------\n",
       "6  -----------------------------------M----------------------------\n",
       "7  -----------------------------------M---------------M------------\n",
       "8  -----------------------------------M----------------------------\n",
       "9  ---M---------------M------------MMMM---------------M------------\n",
       "10 -------------------M---------------M----------------------------\n",
       "11 ---M------------------------------MM---------------M------------\n",
       "12 -----------------------------------M----------------------------\n",
       "13 -----------------------------------M----------------------------\n",
       "14 -----------------------------------M----------------------------\n",
       "15 -----------------------------------M---------------M------------\n",
       "16 -----------------------------------M----------------------------\n",
       "17 --------------------------------M--M---------------M------------\n",
       "18 ---M---------------M---------------M---------------M------------\n",
       "19 ---M-------------------------------M---------------M------------\n",
       "20 -------------------M---------------M----------------------------"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "GENETIC_CODE_TABLE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "202835f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>A</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>C</dt>\n",
       "\t\t<dd>'C'</dd>\n",
       "\t<dt>G</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "\t<dt>T</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>M</dt>\n",
       "\t\t<dd>'AC'</dd>\n",
       "\t<dt>R</dt>\n",
       "\t\t<dd>'AG'</dd>\n",
       "\t<dt>W</dt>\n",
       "\t\t<dd>'AT'</dd>\n",
       "\t<dt>S</dt>\n",
       "\t\t<dd>'CG'</dd>\n",
       "\t<dt>Y</dt>\n",
       "\t\t<dd>'CT'</dd>\n",
       "\t<dt>K</dt>\n",
       "\t\t<dd>'GT'</dd>\n",
       "\t<dt>V</dt>\n",
       "\t\t<dd>'ACG'</dd>\n",
       "\t<dt>H</dt>\n",
       "\t\t<dd>'ACT'</dd>\n",
       "\t<dt>D</dt>\n",
       "\t\t<dd>'AGT'</dd>\n",
       "\t<dt>B</dt>\n",
       "\t\t<dd>'CGT'</dd>\n",
       "\t<dt>N</dt>\n",
       "\t\t<dd>'ACGT'</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[A] 'A'\n",
       "\\item[C] 'C'\n",
       "\\item[G] 'G'\n",
       "\\item[T] 'T'\n",
       "\\item[M] 'AC'\n",
       "\\item[R] 'AG'\n",
       "\\item[W] 'AT'\n",
       "\\item[S] 'CG'\n",
       "\\item[Y] 'CT'\n",
       "\\item[K] 'GT'\n",
       "\\item[V] 'ACG'\n",
       "\\item[H] 'ACT'\n",
       "\\item[D] 'AGT'\n",
       "\\item[B] 'CGT'\n",
       "\\item[N] 'ACGT'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "A\n",
       ":   'A'C\n",
       ":   'C'G\n",
       ":   'G'T\n",
       ":   'T'M\n",
       ":   'AC'R\n",
       ":   'AG'W\n",
       ":   'AT'S\n",
       ":   'CG'Y\n",
       ":   'CT'K\n",
       ":   'GT'V\n",
       ":   'ACG'H\n",
       ":   'ACT'D\n",
       ":   'AGT'B\n",
       ":   'CGT'N\n",
       ":   'ACGT'\n",
       "\n"
      ],
      "text/plain": [
       "     A      C      G      T      M      R      W      S      Y      K      V \n",
       "   \"A\"    \"C\"    \"G\"    \"T\"   \"AC\"   \"AG\"   \"AT\"   \"CG\"   \"CT\"   \"GT\"  \"ACG\" \n",
       "     H      D      B      N \n",
       " \"ACT\"  \"AGT\"  \"CGT\" \"ACGT\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "IUPAC_CODE_MAP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "235bac92",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'A'</li>\n",
       "\t<li>'C'</li>\n",
       "\t<li>'G'</li>\n",
       "\t<li>'U'</li>\n",
       "\t<li>'M'</li>\n",
       "\t<li>'R'</li>\n",
       "\t<li>'W'</li>\n",
       "\t<li>'S'</li>\n",
       "\t<li>'Y'</li>\n",
       "\t<li>'K'</li>\n",
       "\t<li>'V'</li>\n",
       "\t<li>'H'</li>\n",
       "\t<li>'D'</li>\n",
       "\t<li>'B'</li>\n",
       "\t<li>'N'</li>\n",
       "\t<li>'-'</li>\n",
       "\t<li>'+'</li>\n",
       "\t<li>'.'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A'\n",
       "\\item 'C'\n",
       "\\item 'G'\n",
       "\\item 'U'\n",
       "\\item 'M'\n",
       "\\item 'R'\n",
       "\\item 'W'\n",
       "\\item 'S'\n",
       "\\item 'Y'\n",
       "\\item 'K'\n",
       "\\item 'V'\n",
       "\\item 'H'\n",
       "\\item 'D'\n",
       "\\item 'B'\n",
       "\\item 'N'\n",
       "\\item '-'\n",
       "\\item '+'\n",
       "\\item '.'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A'\n",
       "2. 'C'\n",
       "3. 'G'\n",
       "4. 'U'\n",
       "5. 'M'\n",
       "6. 'R'\n",
       "7. 'W'\n",
       "8. 'S'\n",
       "9. 'Y'\n",
       "10. 'K'\n",
       "11. 'V'\n",
       "12. 'H'\n",
       "13. 'D'\n",
       "14. 'B'\n",
       "15. 'N'\n",
       "16. '-'\n",
       "17. '+'\n",
       "18. '.'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"A\" \"C\" \"G\" \"U\" \"M\" \"R\" \"W\" \"S\" \"Y\" \"K\" \"V\" \"H\" \"D\" \"B\" \"N\" \"-\" \"+\" \".\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "RNA_ALPHABET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "97fb5bd3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'A'</li>\n",
       "\t<li>'C'</li>\n",
       "\t<li>'G'</li>\n",
       "\t<li>'U'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'A'\n",
       "\\item 'C'\n",
       "\\item 'G'\n",
       "\\item 'U'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'A'\n",
       "2. 'C'\n",
       "3. 'G'\n",
       "4. 'U'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"A\" \"C\" \"G\" \"U\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "RNA_BASES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ea492517",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl class=dl-horizontal>\n",
       "\t<dt>UUU</dt>\n",
       "\t\t<dd>'F'</dd>\n",
       "\t<dt>UUC</dt>\n",
       "\t\t<dd>'F'</dd>\n",
       "\t<dt>UUA</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>UUG</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>UCU</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>UCC</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>UCA</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>UCG</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>UAU</dt>\n",
       "\t\t<dd>'Y'</dd>\n",
       "\t<dt>UAC</dt>\n",
       "\t\t<dd>'Y'</dd>\n",
       "\t<dt>UAA</dt>\n",
       "\t\t<dd>'*'</dd>\n",
       "\t<dt>UAG</dt>\n",
       "\t\t<dd>'*'</dd>\n",
       "\t<dt>UGU</dt>\n",
       "\t\t<dd>'C'</dd>\n",
       "\t<dt>UGC</dt>\n",
       "\t\t<dd>'C'</dd>\n",
       "\t<dt>UGA</dt>\n",
       "\t\t<dd>'*'</dd>\n",
       "\t<dt>UGG</dt>\n",
       "\t\t<dd>'W'</dd>\n",
       "\t<dt>CUU</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>CUC</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>CUA</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>CUG</dt>\n",
       "\t\t<dd>'L'</dd>\n",
       "\t<dt>CCU</dt>\n",
       "\t\t<dd>'P'</dd>\n",
       "\t<dt>CCC</dt>\n",
       "\t\t<dd>'P'</dd>\n",
       "\t<dt>CCA</dt>\n",
       "\t\t<dd>'P'</dd>\n",
       "\t<dt>CCG</dt>\n",
       "\t\t<dd>'P'</dd>\n",
       "\t<dt>CAU</dt>\n",
       "\t\t<dd>'H'</dd>\n",
       "\t<dt>CAC</dt>\n",
       "\t\t<dd>'H'</dd>\n",
       "\t<dt>CAA</dt>\n",
       "\t\t<dd>'Q'</dd>\n",
       "\t<dt>CAG</dt>\n",
       "\t\t<dd>'Q'</dd>\n",
       "\t<dt>CGU</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>CGC</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>CGA</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>CGG</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>AUU</dt>\n",
       "\t\t<dd>'I'</dd>\n",
       "\t<dt>AUC</dt>\n",
       "\t\t<dd>'I'</dd>\n",
       "\t<dt>AUA</dt>\n",
       "\t\t<dd>'I'</dd>\n",
       "\t<dt>AUG</dt>\n",
       "\t\t<dd>'M'</dd>\n",
       "\t<dt>ACU</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>ACC</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>ACA</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>ACG</dt>\n",
       "\t\t<dd>'T'</dd>\n",
       "\t<dt>AAU</dt>\n",
       "\t\t<dd>'N'</dd>\n",
       "\t<dt>AAC</dt>\n",
       "\t\t<dd>'N'</dd>\n",
       "\t<dt>AAA</dt>\n",
       "\t\t<dd>'K'</dd>\n",
       "\t<dt>AAG</dt>\n",
       "\t\t<dd>'K'</dd>\n",
       "\t<dt>AGU</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>AGC</dt>\n",
       "\t\t<dd>'S'</dd>\n",
       "\t<dt>AGA</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>AGG</dt>\n",
       "\t\t<dd>'R'</dd>\n",
       "\t<dt>GUU</dt>\n",
       "\t\t<dd>'V'</dd>\n",
       "\t<dt>GUC</dt>\n",
       "\t\t<dd>'V'</dd>\n",
       "\t<dt>GUA</dt>\n",
       "\t\t<dd>'V'</dd>\n",
       "\t<dt>GUG</dt>\n",
       "\t\t<dd>'V'</dd>\n",
       "\t<dt>GCU</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>GCC</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>GCA</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>GCG</dt>\n",
       "\t\t<dd>'A'</dd>\n",
       "\t<dt>GAU</dt>\n",
       "\t\t<dd>'D'</dd>\n",
       "\t<dt>GAC</dt>\n",
       "\t\t<dd>'D'</dd>\n",
       "\t<dt>GAA</dt>\n",
       "\t\t<dd>'E'</dd>\n",
       "\t<dt>GAG</dt>\n",
       "\t\t<dd>'E'</dd>\n",
       "\t<dt>GGU</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "\t<dt>GGC</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "\t<dt>GGA</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "\t<dt>GGG</dt>\n",
       "\t\t<dd>'G'</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[UUU] 'F'\n",
       "\\item[UUC] 'F'\n",
       "\\item[UUA] 'L'\n",
       "\\item[UUG] 'L'\n",
       "\\item[UCU] 'S'\n",
       "\\item[UCC] 'S'\n",
       "\\item[UCA] 'S'\n",
       "\\item[UCG] 'S'\n",
       "\\item[UAU] 'Y'\n",
       "\\item[UAC] 'Y'\n",
       "\\item[UAA] '*'\n",
       "\\item[UAG] '*'\n",
       "\\item[UGU] 'C'\n",
       "\\item[UGC] 'C'\n",
       "\\item[UGA] '*'\n",
       "\\item[UGG] 'W'\n",
       "\\item[CUU] 'L'\n",
       "\\item[CUC] 'L'\n",
       "\\item[CUA] 'L'\n",
       "\\item[CUG] 'L'\n",
       "\\item[CCU] 'P'\n",
       "\\item[CCC] 'P'\n",
       "\\item[CCA] 'P'\n",
       "\\item[CCG] 'P'\n",
       "\\item[CAU] 'H'\n",
       "\\item[CAC] 'H'\n",
       "\\item[CAA] 'Q'\n",
       "\\item[CAG] 'Q'\n",
       "\\item[CGU] 'R'\n",
       "\\item[CGC] 'R'\n",
       "\\item[CGA] 'R'\n",
       "\\item[CGG] 'R'\n",
       "\\item[AUU] 'I'\n",
       "\\item[AUC] 'I'\n",
       "\\item[AUA] 'I'\n",
       "\\item[AUG] 'M'\n",
       "\\item[ACU] 'T'\n",
       "\\item[ACC] 'T'\n",
       "\\item[ACA] 'T'\n",
       "\\item[ACG] 'T'\n",
       "\\item[AAU] 'N'\n",
       "\\item[AAC] 'N'\n",
       "\\item[AAA] 'K'\n",
       "\\item[AAG] 'K'\n",
       "\\item[AGU] 'S'\n",
       "\\item[AGC] 'S'\n",
       "\\item[AGA] 'R'\n",
       "\\item[AGG] 'R'\n",
       "\\item[GUU] 'V'\n",
       "\\item[GUC] 'V'\n",
       "\\item[GUA] 'V'\n",
       "\\item[GUG] 'V'\n",
       "\\item[GCU] 'A'\n",
       "\\item[GCC] 'A'\n",
       "\\item[GCA] 'A'\n",
       "\\item[GCG] 'A'\n",
       "\\item[GAU] 'D'\n",
       "\\item[GAC] 'D'\n",
       "\\item[GAA] 'E'\n",
       "\\item[GAG] 'E'\n",
       "\\item[GGU] 'G'\n",
       "\\item[GGC] 'G'\n",
       "\\item[GGA] 'G'\n",
       "\\item[GGG] 'G'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "UUU\n",
       ":   'F'UUC\n",
       ":   'F'UUA\n",
       ":   'L'UUG\n",
       ":   'L'UCU\n",
       ":   'S'UCC\n",
       ":   'S'UCA\n",
       ":   'S'UCG\n",
       ":   'S'UAU\n",
       ":   'Y'UAC\n",
       ":   'Y'UAA\n",
       ":   '*'UAG\n",
       ":   '*'UGU\n",
       ":   'C'UGC\n",
       ":   'C'UGA\n",
       ":   '*'UGG\n",
       ":   'W'CUU\n",
       ":   'L'CUC\n",
       ":   'L'CUA\n",
       ":   'L'CUG\n",
       ":   'L'CCU\n",
       ":   'P'CCC\n",
       ":   'P'CCA\n",
       ":   'P'CCG\n",
       ":   'P'CAU\n",
       ":   'H'CAC\n",
       ":   'H'CAA\n",
       ":   'Q'CAG\n",
       ":   'Q'CGU\n",
       ":   'R'CGC\n",
       ":   'R'CGA\n",
       ":   'R'CGG\n",
       ":   'R'AUU\n",
       ":   'I'AUC\n",
       ":   'I'AUA\n",
       ":   'I'AUG\n",
       ":   'M'ACU\n",
       ":   'T'ACC\n",
       ":   'T'ACA\n",
       ":   'T'ACG\n",
       ":   'T'AAU\n",
       ":   'N'AAC\n",
       ":   'N'AAA\n",
       ":   'K'AAG\n",
       ":   'K'AGU\n",
       ":   'S'AGC\n",
       ":   'S'AGA\n",
       ":   'R'AGG\n",
       ":   'R'GUU\n",
       ":   'V'GUC\n",
       ":   'V'GUA\n",
       ":   'V'GUG\n",
       ":   'V'GCU\n",
       ":   'A'GCC\n",
       ":   'A'GCA\n",
       ":   'A'GCG\n",
       ":   'A'GAU\n",
       ":   'D'GAC\n",
       ":   'D'GAA\n",
       ":   'E'GAG\n",
       ":   'E'GGU\n",
       ":   'G'GGC\n",
       ":   'G'GGA\n",
       ":   'G'GGG\n",
       ":   'G'\n",
       "\n"
      ],
      "text/plain": [
       "UUU UUC UUA UUG UCU UCC UCA UCG UAU UAC UAA UAG UGU UGC UGA UGG CUU CUC CUA CUG \n",
       "\"F\" \"F\" \"L\" \"L\" \"S\" \"S\" \"S\" \"S\" \"Y\" \"Y\" \"*\" \"*\" \"C\" \"C\" \"*\" \"W\" \"L\" \"L\" \"L\" \"L\" \n",
       "CCU CCC CCA CCG CAU CAC CAA CAG CGU CGC CGA CGG AUU AUC AUA AUG ACU ACC ACA ACG \n",
       "\"P\" \"P\" \"P\" \"P\" \"H\" \"H\" \"Q\" \"Q\" \"R\" \"R\" \"R\" \"R\" \"I\" \"I\" \"I\" \"M\" \"T\" \"T\" \"T\" \"T\" \n",
       "AAU AAC AAA AAG AGU AGC AGA AGG GUU GUC GUA GUG GCU GCC GCA GCG GAU GAC GAA GAG \n",
       "\"N\" \"N\" \"K\" \"K\" \"S\" \"S\" \"R\" \"R\" \"V\" \"V\" \"V\" \"V\" \"A\" \"A\" \"A\" \"A\" \"D\" \"D\" \"E\" \"E\" \n",
       "GGU GGC GGA GGG \n",
       "\"G\" \"G\" \"G\" \"G\" \n",
       "attr(,\"alt_init_codons\")\n",
       "[1] \"UUG\" \"CUG\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "RNA_GENETIC_CODE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e857e2cb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
